{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2ac60a33-8726-4b8c-b46e-cd6770101f1a",
   "metadata": {},
   "source": [
    "# Step2-GWASAndIndividualGenotypeDataQualityControls\n",
    "\n",
    "## Note on GWAS File\n",
    "\n",
    "1. It is important to note that different polygenic risk tools accept Genome-Wide Association summary statistic files in various formats.\n",
    "2. The information we have encompasses all the columns required by PRS tools (where some fields are missing, we will highlight it).\n",
    "3. For each individual tool, we will process this GWAS file so that it can be consumed by a specific PRS tool for prediction. The specific number of fields and the names of those fields are highlighted in each tool's documentation.\n",
    "\n",
    "**Fields in the GWAS File:**\n",
    "\n",
    "These fields are also explained in [this link](https://choishingwan.github.io/PRS-Tutorial/base/).\n",
    "\n",
    "\n",
    "| CHR | BP     | SNP        | A1 | A2 | N      | SE        | P         | OR                | INFO             | MAF               |\n",
    "|-----|--------|------------|----|----|--------|-----------|-----------|-------------------|------------------|-------------------|\n",
    "| 1   | 756604 | rs3131962  | A  | G  | 388028 | 0.00301666| 0.483171  | 0.997886915712657 | 0.890557941364774| 0.369389592764921 |\n",
    "| 1   | 768448 | rs12562034 | A  | G  | 388028 | 0.00329472| 0.834808  | 1.00068731609353  | 0.895893511351165| 0.336845754096289 |\n",
    "| 1   | 779322 | rs4040617  | G  | A  | 388028 | 0.00303344| 0.42897   | 0.997603556067569 | 0.897508290615237| 0.377368010940814 |\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4637e743-f111-4923-9440-74f3d632c902",
   "metadata": {},
   "source": [
    " \n",
    "## Genome-Wide Association Study (GWAS) (Base Data) - Quality Controls\n",
    "\n",
    "Please note that some tools have built-in quality control procedures for both GWAS and genotype data.\n",
    "\n",
    "\n",
    "### Set the directory where the files are located\n",
    "\n",
    "```python\n",
    "filedirec = \"SampleData1\"\n",
    "```\n",
    "\n",
    "In case you are running the jobs on HPC and want to use parallel computing, you can replace it with `sys.argv[1]`. All the files for a specific phenotype will be produced in that specific directory, which in this case is `SampleData1`.\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "88a2ff12-16c8-40c7-82ae-5cdfae44b349",
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'SampleData1\\\\SampleData1.gz'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[1], line 19\u001b[0m\n\u001b[0;32m     16\u001b[0m GWAS \u001b[38;5;241m=\u001b[39m filedirec \u001b[38;5;241m+\u001b[39m os\u001b[38;5;241m.\u001b[39msep \u001b[38;5;241m+\u001b[39m filedirec\u001b[38;5;241m+\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m.gz\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m     18\u001b[0m \u001b[38;5;66;03m# Read GWAS data from a compressed file using pandas\u001b[39;00m\n\u001b[1;32m---> 19\u001b[0m df \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[43mGWAS\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mgzip\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\\\u001b[39;49m\u001b[38;5;124;43ms+\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m     21\u001b[0m \u001b[38;5;66;03m# Display the initial number of rows in the dataframe\u001b[39;00m\n\u001b[0;32m     22\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mInitial number of SNPs:\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28mlen\u001b[39m(df))\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\fuck\\Lib\\site-packages\\pandas\\io\\parsers\\readers.py:1026\u001b[0m, in \u001b[0;36mread_csv\u001b[1;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[0;32m   1013\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[0;32m   1014\u001b[0m     dialect,\n\u001b[0;32m   1015\u001b[0m     delimiter,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   1022\u001b[0m     dtype_backend\u001b[38;5;241m=\u001b[39mdtype_backend,\n\u001b[0;32m   1023\u001b[0m )\n\u001b[0;32m   1024\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[1;32m-> 1026\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\fuck\\Lib\\site-packages\\pandas\\io\\parsers\\readers.py:620\u001b[0m, in \u001b[0;36m_read\u001b[1;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[0;32m    617\u001b[0m _validate_names(kwds\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnames\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[0;32m    619\u001b[0m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[1;32m--> 620\u001b[0m parser \u001b[38;5;241m=\u001b[39m \u001b[43mTextFileReader\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    622\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[0;32m    623\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\fuck\\Lib\\site-packages\\pandas\\io\\parsers\\readers.py:1620\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[1;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[0;32m   1617\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m kwds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[0;32m   1619\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles: IOHandles \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m-> 1620\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_engine\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mengine\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\fuck\\Lib\\site-packages\\pandas\\io\\parsers\\readers.py:1880\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[1;34m(self, f, engine)\u001b[0m\n\u001b[0;32m   1878\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m mode:\n\u001b[0;32m   1879\u001b[0m         mode \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m-> 1880\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;241m=\u001b[39m \u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m   1881\u001b[0m \u001b[43m    \u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1882\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1883\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1884\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcompression\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1885\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmemory_map\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmemory_map\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1886\u001b[0m \u001b[43m    \u001b[49m\u001b[43mis_text\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mis_text\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1887\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding_errors\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstrict\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1888\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstorage_options\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1889\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1890\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m   1891\u001b[0m f \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles\u001b[38;5;241m.\u001b[39mhandle\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\fuck\\Lib\\site-packages\\pandas\\io\\common.py:765\u001b[0m, in \u001b[0;36mget_handle\u001b[1;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[0;32m    761\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m compression \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgzip\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m    762\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(handle, \u001b[38;5;28mstr\u001b[39m):\n\u001b[0;32m    763\u001b[0m         \u001b[38;5;66;03m# error: Incompatible types in assignment (expression has type\u001b[39;00m\n\u001b[0;32m    764\u001b[0m         \u001b[38;5;66;03m# \"GzipFile\", variable has type \"Union[str, BaseBuffer]\")\u001b[39;00m\n\u001b[1;32m--> 765\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[43mgzip\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mGzipFile\u001b[49m\u001b[43m(\u001b[49m\u001b[43m  \u001b[49m\u001b[38;5;66;43;03m# type: ignore[assignment]\u001b[39;49;00m\n\u001b[0;32m    766\u001b[0m \u001b[43m            \u001b[49m\u001b[43mfilename\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    767\u001b[0m \u001b[43m            \u001b[49m\u001b[43mmode\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    768\u001b[0m \u001b[43m            \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mcompression_args\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    769\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    770\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    771\u001b[0m         handle \u001b[38;5;241m=\u001b[39m gzip\u001b[38;5;241m.\u001b[39mGzipFile(\n\u001b[0;32m    772\u001b[0m             \u001b[38;5;66;03m# No overload variant of \"GzipFile\" matches argument types\u001b[39;00m\n\u001b[0;32m    773\u001b[0m             \u001b[38;5;66;03m# \"Union[str, BaseBuffer]\", \"str\", \"Dict[str, Any]\"\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    776\u001b[0m             \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mcompression_args,\n\u001b[0;32m    777\u001b[0m         )\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\fuck\\Lib\\gzip.py:174\u001b[0m, in \u001b[0;36mGzipFile.__init__\u001b[1;34m(self, filename, mode, compresslevel, fileobj, mtime)\u001b[0m\n\u001b[0;32m    172\u001b[0m     mode \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m    173\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m fileobj \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m--> 174\u001b[0m     fileobj \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmyfileobj \u001b[38;5;241m=\u001b[39m \u001b[43mbuiltins\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mopen\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilename\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01mor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mrb\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m    175\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m filename \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m    176\u001b[0m     filename \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mgetattr\u001b[39m(fileobj, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mname\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'SampleData1\\\\SampleData1.gz'"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import subprocess\n",
    "\n",
    "# Set the directory where the files are located\n",
    "filedirec = \"SampleData1\"\n",
    "#filedirec = \"asthma_19\"\n",
    "#filedirec = \"migraine_0\"\n",
    "\n",
    "# Define file paths for different data files\n",
    "BED = filedirec + os.sep + filedirec\n",
    "BIM = filedirec + os.sep + filedirec+\".bim\"\n",
    "FAM = filedirec + os.sep + filedirec+\".fam\"\n",
    "COV = filedirec + os.sep + filedirec+\".cov\"\n",
    "Height = filedirec + os.sep + filedirec+\".height\"\n",
    "GWAS = filedirec + os.sep + filedirec+\".gz\"\n",
    "\n",
    "# Read GWAS data from a compressed file using pandas\n",
    "df = pd.read_csv(GWAS, compression=\"gzip\", sep=\"\\s+\")\n",
    "\n",
    "# Display the initial number of rows in the dataframe\n",
    "print(\"Initial number of SNPs:\", len(df))\n",
    "\n",
    "# Apply quality control steps: Filter SNPs based on Minor Allele Frequency (MAF) and Imputation Information Score (INFO)\n",
    "df = df.loc[(df['MAF'] > 0.01) & (df['INFO'] > 0.8)]\n",
    "\n",
    "# Display the number of rows after applying the filters\n",
    "print(\"Number of SNPs after quality control:\", len(df))\n",
    "\n",
    " \n",
    "# Display the number of rows after removing duplicate SNPs\n",
    "print(\"SNPs in GWAS after removing duplicate SNPs:\", len(df))\n",
    "\n",
    "# Remove ambiguous SNPs with complementary alleles (C/G or A/T) to avoid potential errors\n",
    "df = df[~((df['A1'] == 'A') & (df['A2'] == 'T') |\n",
    "          (df['A1'] == 'T') & (df['A2'] == 'A') |\n",
    "          (df['A1'] == 'G') & (df['A2'] == 'C') |\n",
    "          (df['A1'] == 'C') & (df['A2'] == 'G'))]\n",
    "\n",
    "# Display the final number of SNPs after removing ambiguous SNPs\n",
    "print(\"Final number of SNPs after removing ambiguous SNPs:\", len(df))\n",
    "\n",
    "# Save the data.\n",
    "df.to_csv(GWAS,compression=\"gzip\",sep=\"\\t\",index=None)\n",
    "\n",
    "df = pd.read_csv(GWAS,compression= \"gzip\",sep=\"\\s+\")\n",
    "print(len(df))\n",
    "print(df.head().to_markdown())\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02050bdc-e82d-4b91-b8be-71f89321fbcc",
   "metadata": {},
   "source": [
    "## Match Variants Between GWAS and Individual Genotype Data\n",
    "\n",
    "If RSID is present in the GWAS, the following step can be skipped.\n",
    "\n",
    "### Steps for Handling RSIDs in GWAS and Genotype Data\n",
    "\n",
    "1. If RSIDs are not present for SNPs, put `X` in the SNP column in the GWAS file.\n",
    "\n",
    "2. Read the `genotype.bim` file and extract the RSIDs from the genotype data.\n",
    "\n",
    "3. If RSIDs are not present in the genotype data, use HapMap3 or another reference panel to obtain the RSIDs.\n",
    "\n",
    "\n",
    "Some PRS tools use different criteria to create unique variants and match them between GWAS and individual genotype data:\n",
    "\n",
    "- **CHR:BP:A1:A2**: Some PRS tools use this format to define a unique variant.\n",
    "- **RSID**: Some PRS tools use RSID/SNP to define a unique variant.\n",
    "- **CHR:BP**: Some PRS tools use this format to define a unique variant.\n",
    "\n",
    "We have highlighted which criteria are necessary for each tool.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "9ef16b48-0e8d-4343-b018-02eae66faff8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Columns of BIM file:\n",
      "Index([0, 1, 2, 3, 4, 5], dtype='int64')\n",
      "First 10 rows of BIM file:\n",
      "Removing SNPs for which even a single row does not contain the required value: 499617\n",
      "RSID is present!\n",
      "Total SNPs 499617\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "bimfile = pd.read_csv(BIM, sep=\"\\s+\", header=None)\n",
    "print(\"Columns of BIM file:\")\n",
    "print(bimfile.columns)\n",
    "print(\"First 10 rows of BIM file:\")\n",
    "\n",
    "\n",
    "print(\"Removing SNPs for which even a single row does not contain the required value:\", len(df))\n",
    "\n",
    "\n",
    "# If RSID's are not present for SNPs, put X in the SNP column in the GWAS file.\n",
    "# Read the genotype.bim file, and extract the RSID from the genotype data.\n",
    "# If RSID are not present in the genotype data, use HapMap3 or other reference panel to get the RSIDs.\n",
    "\n",
    "if (df['SNP'] == 'X').all():\n",
    "    print(\"RSIDs are missing!\")\n",
    "    bimfile = pd.read_csv(filedirec+os.sep+filedirec+\".bim\", sep=\"\\s+\", header=None)\n",
    "    \n",
    "    # create a unique variant using CHR:BP:A1:A2.\n",
    "    \n",
    "    bimfile[\"match\"] = bimfile[0].astype(str)+\"_\"+bimfile[3].astype(str)+\"_\"+bimfile[4].astype(str)+\"_\"+bimfile[5].astype(str)\n",
    "    df[\"match\"] = df[\"CHR\"].astype(str)+\"_\"+df[\"BP\"].astype(str)+\"_\"+df[\"A1\"].astype(str)+\"_\"+df[\"A2\"].astype(str)\n",
    "    \n",
    "\n",
    "  \n",
    "\n",
    "    df.drop_duplicates(subset='match', inplace=True)\n",
    "    bimfile.drop_duplicates(subset='match', inplace=True)\n",
    "\n",
    "    df = df[df['match'].isin(bimfile['match'].values)]\n",
    "    bimfile = bimfile[bimfile['match'].isin(df['match'].values)]\n",
    "    df = df[df['match'].isin(bimfile['match'].values)]\n",
    "    bimfile = bimfile[bimfile['match'].isin(df['match'].values)]\n",
    " \n",
    "    \n",
    "    df = df.sort_values(by='BP')\n",
    "    bimfile = bimfile.sort_values(by=3)\n",
    "    \n",
    "    print(df.head())\n",
    "    print(bimfile.head())\n",
    "\n",
    "    df[\"SNP\"] = bimfile[1].values\n",
    "    print(\"match\",len(df))\n",
    "\n",
    "\n",
    "    df.drop_duplicates(subset='match', inplace=True)\n",
    "    bimfile.drop_duplicates(subset='match', inplace=True)  \n",
    "\n",
    "    print(len(df))\n",
    "    print(len(bimfile))\n",
    "    print(df.head())\n",
    "    print(bimfile.head())\n",
    "    \n",
    "    del df[\"match\"]\n",
    "    # Just save the modified GWAS file.\n",
    "    # If bim, file is modified, the genotype data will be considered as corupt by Plink.\n",
    "    df.to_csv(GWAS,compression=\"gzip\",sep=\"\\t\",index=None)   \n",
    "    print(\"Total SNPs\", len(df))\n",
    "\n",
    "    pass\n",
    "else:\n",
    "    df.drop_duplicates(subset='SNP', inplace=True)\n",
    "    df.to_csv(GWAS,compression=\"gzip\",sep=\"\\t\",index=None)\n",
    "    print(\"RSID is present!\")\n",
    "    print(\"Total SNPs\",len(df))\n",
    "    pass\n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e0c651a-70ae-4997-a463-4ae5d31e02a9",
   "metadata": {},
   "source": [
    "## Individual genotype data (Target Data) Processing\n",
    "\n",
    "Ensure that the phenotype file, FAM file, and covariate file contain an identical number of samples. Remove any missing samples based on your data. Note that the extent of missingness in phenotypes and covariates may vary.\n",
    "\n",
    "\n",
    "**Note:** Plink needs to be installed or placed in the same directory as this notebook.\n",
    "\n",
    "[Download Plink](https://www.cog-genomics.org/plink/)\n",
    "\n",
    "We recommend using Linux. In cases where Windows is required due to package installation issues on Linux, we provide the following guidance:\n",
    "\n",
    "1. For Windows, use `plink`.\n",
    "2. For Linux, use `./plink`.\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9372ccb9-a643-44e7-a3eb-d05eb691834e",
   "metadata": {},
   "source": [
    "### Remove people with missing Phenotype\n",
    "\n",
    "Modify the fam file, make bed file, and modify the covariates files as well."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c9aef7cf-3109-4cca-9c30-0a20393562cc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FAM file contents:\n",
      "       FID      IID  Father  Mother  Sex  Phenotype\n",
      "0  HG00096  HG00096       0       0    1         -9\n",
      "1  HG00097  HG00097       0       0    2         -9\n",
      "2  HG00099  HG00099       0       0    2         -9\n",
      "3  HG00100  HG00100       0       0    2         -9\n",
      "4  HG00101  HG00101       0       0    1         -9\n",
      "Total number of people in FAM file: 503\n",
      "Phenotype information is available for: 475 people\n",
      "475\n",
      "         FID      IID  Father  Mother  Sex   Phenotype\n",
      "0    HG00096  HG00096       0       0    1  169.132169\n",
      "1    HG00097  HG00097       0       0    2  171.256259\n",
      "2    HG00099  HG00099       0       0    2  171.534380\n",
      "3    HG00101  HG00101       0       0    1  169.850176\n",
      "4    HG00102  HG00102       0       0    2  172.788361\n",
      "..       ...      ...     ...     ...  ...         ...\n",
      "470  NA20822  NA20822       0       0    2  170.405056\n",
      "471  NA20826  NA20826       0       0    2  168.523029\n",
      "472  NA20827  NA20827       0       0    1  170.975735\n",
      "473  NA20828  NA20828       0       0    2  170.222028\n",
      "474  NA20832  NA20832       0       0    2  169.431705\n",
      "\n",
      "[475 rows x 6 columns]\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/SampleData1_QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/SampleData1\n",
      "  --keep SampleData1/PeopleWithPhenotype.txt\n",
      "  --make-bed\n",
      "  --out SampleData1/SampleData1_QC\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "503 people (240 males, 263 females) loaded from .fam.\n",
      "--keep: 475 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 475 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999896.\n",
      "551892 variants and 475 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to SampleData1/SampleData1_QC.bed + SampleData1/SampleData1_QC.bim +\n",
      "SampleData1/SampleData1_QC.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Covariate file contents:\n",
      "       FID      IID  Sex\n",
      "0  HG00096  HG00096    1\n",
      "1  HG00097  HG00097    2\n",
      "2  HG00099  HG00099    2\n",
      "3  HG00100  HG00100    2\n",
      "4  HG00101  HG00101    1\n",
      "Total number of people in Covariate file: 503\n",
      "Covariate file contents after matching with FAM file:\n",
      "       FID      IID  Sex\n",
      "0  HG00096  HG00096    1\n",
      "1  HG00097  HG00097    2\n",
      "2  HG00099  HG00099    2\n",
      "4  HG00101  HG00101    1\n",
      "5  HG00102  HG00102    2\n",
      "Total number of people in Covariate file after matching: 475\n"
     ]
    }
   ],
   "source": [
    "# New files to be saved with QC suffix\n",
    "newfilename = filedirec + \"_QC\"\n",
    "\n",
    "# Read information from FAM file\n",
    "f = pd.read_csv(FAM, header=None, sep=\"\\s+\", names=[\"FID\", \"IID\", \"Father\", \"Mother\", \"Sex\", \"Phenotype\"])\n",
    "print(\"FAM file contents:\")\n",
    "print(f.head())\n",
    "print(\"Total number of people in FAM file:\", len(f))\n",
    "\n",
    "# Append the Height phenotype values to FAM file\n",
    "# Height file is basically the phenotype file.\n",
    "h = pd.read_csv(Height, sep=\"\\t\")\n",
    "print(\"Phenotype information is available for:\", len(h), \"people\")\n",
    "print(len(h))\n",
    "result = pd.merge(f, h, on=['FID', 'IID'])\n",
    "\n",
    "# Replace 'Phenotype' column with 'Height' and save to a new PeopleWithPhenotype.txt file\n",
    "# Ensure that the input Phenotype file has teh header Height.\n",
    "result[\"Phenotype\"] = result[\"Height\"].values\n",
    "del result[\"Height\"]\n",
    "\n",
    "# Remove NA or missing in the phenotype column\n",
    "result = result.dropna(subset=[\"Phenotype\"])\n",
    "\n",
    "\n",
    "print(result)\n",
    "result.to_csv(filedirec + os.sep + \"PeopleWithPhenotype.txt\", index=False, header=False, sep=\"\\t\")\n",
    "\n",
    "# Use plink to keep only the people with phenotype present\n",
    "plink_command = [\n",
    "    './plink',\n",
    "    '--bfile', filedirec + os.sep + filedirec,\n",
    "    '--keep', filedirec + os.sep + \"PeopleWithPhenotype.txt\",\n",
    "    '--make-bed',\n",
    "    '--out', filedirec + os.sep + newfilename\n",
    "]\n",
    "subprocess.run(plink_command)\n",
    "\n",
    "# Update the phenotype information in the new FAM file\n",
    "f = pd.read_csv(filedirec + os.sep + newfilename + \".fam\", header=None, sep=\"\\s+\",\n",
    "                names=[\"FID\", \"IID\", \"Father\", \"Mother\", \"Sex\", \"Phenotype\"])\n",
    "f[\"Phenotype\"] = result[\"Phenotype\"].values\n",
    "f.to_csv(filedirec + os.sep + newfilename + \".fam\", index=False, header=False, sep=\"\\t\")\n",
    "\n",
    "# Update the covariate file as well\n",
    "covfile = filedirec + os.sep + filedirec + '.cov'\n",
    "covfile = pd.read_csv(covfile, sep=\"\\s+\")\n",
    "\n",
    "print(\"Covariate file contents:\")\n",
    "print(covfile.head())\n",
    "print(\"Total number of people in Covariate file:\", len(covfile))\n",
    "\n",
    "# Match the FID and IID from covariate and height file\n",
    "covfile = covfile[covfile['FID'].isin(f[\"FID\"].values) & covfile['IID'].isin(f[\"IID\"].values)]\n",
    "print(\"Covariate file contents after matching with FAM file:\")\n",
    "print(covfile.head())\n",
    "print(\"Total number of people in Covariate file after matching:\", len(covfile))\n",
    "covfile.to_csv(filedirec + os.sep + newfilename + \".cov\", index=None, sep=\"\\t\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2837be19-131d-4b4a-8162-00a0a7334b2c",
   "metadata": {},
   "source": [
    "## Split Data into Test and Train and Perform Quality Controls on Training Data\n",
    "\n",
    "1. We adopt a cross-validation design to evaluate polygenic risk scores.\n",
    "2. The base data is divided into two sets: training and test sets. The training set is used to find the best combination of parameters or hyperparameters offered by each tool, along with the summary statistic file. The data is split into 5 folds, and further processing is performed on the first fold. `Fold_0`\n",
    "3. Data is divided into training and test sets, and hyperparameter optimization is performed on the training data. Operations such as clumping or pruning are applied to the training data, and the same remaining SNPs should be extracted from the test set rather than separately using pruning or clumping on test data.\n",
    "4. Regarding hyperparameters from individual tools, all of those should be applied to the training set, and the performance should be measured for both the null model and the complete model (including covariates and polygenic risk scores). Since it is a continuous phenotype, Explained Variance is considered to assess the performance of the PRS model. For binary phenotypes, we considered AUC to evaluate the PRS performance.\n",
    "\n",
    "**Note:** We will divide the data into training and test sets, perform quality controls on the training data. The data for each fold will be saved separately for further processing.\n",
    ".\n",
    "\n",
    "### Pruning\n",
    "Pruning is an integral part of the analysis, but it has been skipped as a quality control on the training set for the following reasons. In our initial analysis, we observed that pruning and clumping can affect the performance of the polygenic risk score model. Rather than performing it at this stage, we consider it as one of the hyperparameters and it will be performed at a later stage. If we perform pruning at this stage, the process that passed the pruning step would limit us to use other values for pruning at the latest stage. However, even at the latest stage, pruning is only performed on the training set.\n",
    "\n",
    "### Cross-validation Designs\n",
    "There are multiple ways of doing cross-validation design, and one of them is to use all the cohorts except one for the training and then use the last cohort as the validation set. We performed quality controls and hyperparameter optimization on the training set and found the best combination across all folds, reporting the performance of the best hyperparameter combination on the test set.\n",
    "\n",
    "### A Simple Analysis\n",
    "If you have a separate GWAS file and our training data that you will be using to optimize the hyperparameter on a subset without cross-validation, then follow the original tutorial presented in the first cell [Shing Wan Choi's PRS Tutorial](https://choishingwan.github.io/PRS-Tutorial/base/).\n",
    "\n",
    "**Quality Controls considered for Training Data:**\n",
    "1. GWAS studies, e.g., removing SNPs with low genotyping rate, low minor allele frequency, out of Hardy-Weinberg Equilibrium, removing individuals with low genotyping rate.\n",
    "2. Pruning was skipped.\n",
    "3. Heterozygosity check.\n",
    "4. Sex chromosomes.\n",
    "5. Relatedness.\n",
    "\n",
    "**R Script - Module1.R**\n",
    "\n",
    "This file contains the code presented in this [tutorial](https://choishingwan.github.io/PRS-Tutorial/base/) to assist in performing quality controls on the training data. Kindly, follow their instructions for better understanding, as quality controls are not the main focus of this research.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0aba3c01",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<a href='Module1.R' target='_blank'>Module1.R</a><br>"
      ],
      "text/plain": [
       "/data/ascher01/uqmmune1/BenchmarkingPGSTools/Module1.R"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from IPython.display import FileLink\n",
    "\n",
    "# R file used in to execute the following code.\n",
    "FileLink('Module1.R')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "94e89b18-ddf5-4b23-bc73-7dd5c8661812",
   "metadata": {},
   "source": [
    "**R Script - Module1.R**"
   ]
  },
  {
   "cell_type": "raw",
   "id": "5fb54437-dff5-47fa-8020-38718e69cfa6",
   "metadata": {},
   "source": [
    " \n",
    "This file contains the code presented in this [tutorial](https://choishingwan.github.io/PRS-Tutorial/base/) to assist in performing quality controls on the training data. Kindly, follow their instructions for better understanding, as quality controls are not the main focus of this research.\n",
    "\n",
    "args <- commandArgs(trailingOnly = TRUE)\n",
    "print(args)\n",
    "# Argument one is going to be the directory.\n",
    "# Argument two is going to be the file name.\n",
    "# Argument three is going to be the output file name.\n",
    "# Argument four is going to be the specific function to be called.\n",
    "\n",
    "if (args[4]==\"1\"){\n",
    "\n",
    "  result <-paste(\".\",args[1],paste(args[3],toString(\".het\"), sep = \"\"),sep=\"//\")\n",
    "  dat <- read.table(result, header=T) # Read in the EUR.het file, specify it has header\n",
    "  m <- mean(dat$F) # Calculate the mean  \n",
    "  s <- sd(dat$F) # Calculate the SD\n",
    "  valid <- subset(dat, F <= m+3*s & F >= m-3*s) # Get any samples with F coefficient within 3 SD of the population mean\n",
    "  result <-paste(\"./\",args[1],paste(args[2],toString(\".valid.sample\"), sep = \"\"),sep=\"//\")\n",
    "  write.table(valid[,c(1,2)], result, quote=F, row.names=F) # print FID and IID for valid samples\n",
    "  result <-paste(\"./\",args[1],paste(args[2],toString(\".bim\"), sep = \"\"),sep=\"//\")\n",
    "  bim <- read.table(result)\n",
    "  \n",
    "  colnames(bim) <- c(\"CHR\", \"SNP\", \"CM\", \"BP\", \"B.A1\", \"B.A2\")\n",
    "  \n",
    "  # Read in QCed SNPs\n",
    "  result <-paste(\"./\",args[1],paste(args[3],toString(\".snplist\"), sep = \"\"),sep=\"//\")\n",
    "  \n",
    "  qc <- read.table(result, header = F, stringsAsFactors = F)\n",
    "  # Read in the GWAS data\n",
    "  path_parts <- strsplit(args[1], \"/|\\\\\\\\\")[[1]]\n",
    "  print(path_parts)\n",
    "  print(path_parts[1])\n",
    "  result <-paste(\"./\",path_parts[1],paste(path_parts[1],toString(\".gz\"), sep = \"\"),sep=\"//\")\n",
    "  #result <-paste(\"./\",args[1],\"train\",toString(\"train.assoc.fisher\"),sep=\"//\")\n",
    "  \n",
    "  height <-read.table(gzfile(result),\n",
    "               header = T,\n",
    "               stringsAsFactors = F, \n",
    "               sep=\"\\t\")\n",
    "  # Change all alleles to upper case for easy comparison\n",
    "  \n",
    "\n",
    "  height$A1 <- toupper(height$A1)\n",
    "  height$A2 <- toupper(height$A2)\n",
    "  bim$B.A1 <- toupper(bim$B.A1)\n",
    "  bim$B.A2 <- toupper(bim$B.A2)\n",
    "  info <- merge(bim, height, by = c(\"SNP\", \"CHR\", \"BP\"))\n",
    "  # Filter QCed SNPs\n",
    "  print(length(info))\n",
    "\n",
    "  info <- info[info$SNP %in% qc$V1,]\n",
    "  print(length(info))\n",
    "  # Function for finding the complementary allele\n",
    "  \n",
    "  complement <- function(x) {\n",
    "    switch (\n",
    "      x,\n",
    "      \"A\" = \"T\",\n",
    "      \"C\" = \"G\",\n",
    "      \"T\" = \"A\",\n",
    "      \"G\" = \"C\",\n",
    "      return(NA)\n",
    "    )\n",
    "  }\n",
    "  \n",
    "  # Get SNPs that have the same alleles across base and target\n",
    "  info.match <- subset(info, A1 == B.A1 & A2 == B.A2)\n",
    "  print(length(info.match))\n",
    "   \n",
    "  # Identify SNPs that are complementary between base and target\n",
    "  info$C.A1 <- sapply(info$B.A1, complement)\n",
    "  info$C.A2 <- sapply(info$B.A2, complement)\n",
    "  info.complement <- subset(info, A1 == C.A1 & A2 == C.A2)\n",
    "  # Update the complementary alleles in the bim file\n",
    "  # This allow us to match the allele in subsequent analysis\n",
    "  \n",
    "  complement.snps <- bim$SNP %in% info.complement$SNP\n",
    "  bim[complement.snps,]$B.A1 <-\n",
    "    sapply(bim[complement.snps,]$B.A1, complement)\n",
    "  bim[complement.snps,]$B.A2 <-\n",
    "    sapply(bim[complement.snps,]$B.A2, complement)\n",
    "  \n",
    "  # identify SNPs that need recoding\n",
    "  info.recode <- subset(info, A1 == B.A2 & A2 == B.A1)\n",
    "  # Update the recode SNPs\n",
    "  recode.snps <- bim$SNP %in% info.recode$SNP\n",
    "  tmp <- bim[recode.snps,]$B.A1\n",
    "  bim[recode.snps,]$B.A1 <- bim[recode.snps,]$B.A2\n",
    "  bim[recode.snps,]$B.A2 <- tmp\n",
    "  \n",
    "  # identify SNPs that need recoding & complement\n",
    "  info.crecode <- subset(info, A1 == C.A2 & A2 == C.A1)\n",
    "  # Update the recode + strand flip SNPs\n",
    "  com.snps <- bim$SNP %in% info.crecode$SNP\n",
    "  tmp <- bim[com.snps,]$B.A1\n",
    "  bim[com.snps,]$B.A1 <- as.character(sapply(bim[com.snps,]$B.A2, complement))\n",
    "  bim[com.snps,]$B.A2 <- as.character(sapply(tmp, complement))\n",
    "  result <-paste(\"./\",args[1],paste(args[2],toString(\".a1\"), sep = \"\"),sep=\"//\")\n",
    "  \n",
    "  # Output updated bim file\n",
    "  write.table(\n",
    "    bim[,c(\"SNP\", \"B.A1\")],\n",
    "    result,\n",
    "    quote = F,\n",
    "    row.names = F,\n",
    "    col.names = F,\n",
    "    sep=\"\\t\"\n",
    "  )\n",
    "  mismatch <-\n",
    "    bim$SNP[!(bim$SNP %in% info.match$SNP |\n",
    "                bim$SNP %in% info.complement$SNP | \n",
    "                bim$SNP %in% info.recode$SNP |\n",
    "                bim$SNP %in% info.crecode$SNP)]\n",
    "  result <-paste(\"./\",args[1],paste(args[2],toString(\".mismatch\"), sep = \"\"),sep=\"//\")\n",
    "  \n",
    "  write.table(\n",
    "    mismatch,\n",
    "    result,\n",
    "    quote = F,\n",
    "    row.names = F,\n",
    "    col.names = F\n",
    "  )\n",
    "  \n",
    "  \n",
    "}\n",
    "if (args[4]==\"2\"){\n",
    "  result <-paste(\"./\",args[1],paste(args[2],toString(\".valid.sample\"), sep = \"\"),sep=\"//\")\n",
    "  valid <- read.table(result, header=T)\n",
    "  result <-paste(\"./\",args[1],paste(args[3],toString(\".sexcheck\"), sep = \"\"),sep=\"//\")\n",
    "  dat <- read.table(result, header=T)\n",
    "  valid <- subset(dat, STATUS==\"OK\" & FID %in% valid$FID)\n",
    "  result <-paste(\"./\",args[1],paste(args[3],toString(\".valid\"), sep = \"\"),sep=\"//\")\n",
    "  print(result)\n",
    "  write.table(valid[,c(\"FID\", \"IID\")], result, row.names=F, col.names=F, sep=\"\\t\", quote=F) \n",
    "}\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e321c60c-65c0-4288-a7a3-aad4641092fa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The column does not contain only 0 and 1.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_0/train_data.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/SampleData1_QC\n",
      "  --keep SampleData1/Fold_0/train_data.fam\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_0/train_data\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "475 people (227 males, 248 females) loaded from .fam.\n",
      "475 phenotype values loaded from .fam.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999896.\n",
      "551892 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_0/train_data.bed +\n",
      "SampleData1/Fold_0/train_data.bim + SampleData1/Fold_0/train_data.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_0/test_data.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/SampleData1_QC\n",
      "  --keep SampleData1/Fold_0/test_data.fam\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_0/test_data\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "475 people (227 males, 248 females) loaded from .fam.\n",
      "475 phenotype values loaded from .fam.\n",
      "--keep: 95 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 95 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999896.\n",
      "551892 variants and 95 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_0/test_data.bed +\n",
      "SampleData1/Fold_0/test_data.bim + SampleData1/Fold_0/test_data.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_0/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_0/train_data\n",
      "  --geno 0.1\n",
      "  --hwe 1e-6\n",
      "  --maf 0.01\n",
      "  --make-just-fam\n",
      "  --mind 0.1\n",
      "  --out SampleData1/Fold_0/train_data.QC\n",
      "  --write-snplist\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (183 males, 197 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "0 people removed due to missing genotype data (--mind).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999896.\n",
      "1 variant removed due to missing genotype data (--geno).\n",
      "--hwe: 822 variants removed due to Hardy-Weinberg exact test.\n",
      "8415 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "542654 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "List of variant IDs written to SampleData1/Fold_0/train_data.QC.snplist .\n",
      "--make-just-fam to SampleData1/Fold_0/train_data.QC.fam ... done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --hwe observation counts vary by more than 10%, due to the X\n",
      "chromosome.  You may want to use a less stringent --hwe p-value threshold for X\n",
      "chromosome variants.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_0/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_0/train_data\n",
      "  --extract SampleData1/Fold_0/train_data.QC.snplist\n",
      "  --het\n",
      "  --keep SampleData1/Fold_0/train_data.QC.fam\n",
      "  --out SampleData1/Fold_0/train_data.QC\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (183 males, 197 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 542654 variants remaining.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999898.\n",
      "542654 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--het: 521371 variants scanned, report written to\n",
      "SampleData1/Fold_0/train_data.QC.het .\n",
      "[1] \"SampleData1/Fold_0\" \"train_data\"         \"train_data.QC\"     \n",
      "[4] \"1\"                 \n",
      "[1] \"SampleData1\" \"Fold_0\"     \n",
      "[1] \"SampleData1\"\n",
      "[1] 14\n",
      "[1] 14\n",
      "[1] 14\n",
      "Rscript Module1.R SampleData1/Fold_0  train_data train_data.QC 1\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_0/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_0/train_data\n",
      "  --extract SampleData1/Fold_0/train_data.QC.snplist\n",
      "  --out SampleData1/Fold_0/train_data.QC\n",
      "  --rel-cutoff 0.125\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (183 males, 197 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 542654 variants remaining.\n",
      "Using up to 8 threads (change this with --threads).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999898.\n",
      "542654 variants and 380 people pass filters and QC (before --rel-cutoff).\n",
      "Phenotype data is quantitative.\n",
      "Excluding 21283 variants on non-autosomes from relationship matrix calc.\n",
      "Relationship matrix calculation complete.\n",
      "0 people excluded by --rel-cutoff.\n",
      "Remaining sample IDs written to SampleData1/Fold_0/train_data.QC.rel.id .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_0/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --a1-allele SampleData1/Fold_0/train_data.a1\n",
      "  --bfile SampleData1/Fold_0/train_data\n",
      "  --exclude SampleData1/Fold_0/train_data.mismatch\n",
      "  --extract SampleData1/Fold_0/train_data.QC.snplist\n",
      "  --keep SampleData1/Fold_0/train_data.QC.rel.id\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_0/train_data.QC\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (183 males, 197 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 542654 variants remaining.\n",
      "--exclude: 491952 variants remaining.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999894.\n",
      "--a1-allele: 491952 assignments made.\n",
      "491952 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_0/train_data.QC.bed +\n",
      "SampleData1/Fold_0/train_data.QC.bim + SampleData1/Fold_0/train_data.QC.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_1/train_data.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/SampleData1_QC\n",
      "  --keep SampleData1/Fold_1/train_data.fam\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_1/train_data\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "475 people (227 males, 248 females) loaded from .fam.\n",
      "475 phenotype values loaded from .fam.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999922.\n",
      "551892 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_1/train_data.bed +\n",
      "SampleData1/Fold_1/train_data.bim + SampleData1/Fold_1/train_data.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_1/test_data.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/SampleData1_QC\n",
      "  --keep SampleData1/Fold_1/test_data.fam\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_1/test_data\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "475 people (227 males, 248 females) loaded from .fam.\n",
      "475 phenotype values loaded from .fam.\n",
      "--keep: 95 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 95 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999794.\n",
      "551892 variants and 95 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_1/test_data.bed +\n",
      "SampleData1/Fold_1/test_data.bim + SampleData1/Fold_1/test_data.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_1/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_1/train_data\n",
      "  --geno 0.1\n",
      "  --hwe 1e-6\n",
      "  --maf 0.01\n",
      "  --make-just-fam\n",
      "  --mind 0.1\n",
      "  --out SampleData1/Fold_1/train_data.QC\n",
      "  --write-snplist\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (178 males, 202 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "0 people removed due to missing genotype data (--mind).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999922.\n",
      "1 variant removed due to missing genotype data (--geno).\n",
      "--hwe: 855 variants removed due to Hardy-Weinberg exact test.\n",
      "7923 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "543113 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "List of variant IDs written to SampleData1/Fold_1/train_data.QC.snplist .\n",
      "--make-just-fam to SampleData1/Fold_1/train_data.QC.fam ... done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_1/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_1/train_data\n",
      "  --extract SampleData1/Fold_1/train_data.QC.snplist\n",
      "  --het\n",
      "  --keep SampleData1/Fold_1/train_data.QC.fam\n",
      "  --out SampleData1/Fold_1/train_data.QC\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (178 males, 202 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 543113 variants remaining.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697%"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --hwe observation counts vary by more than 10%, due to the X\n",
      "chromosome.  You may want to use a less stringent --hwe p-value threshold for X\n",
      "chromosome variants.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\b989 done.\n",
      "Total genotyping rate is 0.999923.\n",
      "543113 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--het: 521800 variants scanned, report written to\n",
      "SampleData1/Fold_1/train_data.QC.het .\n",
      "[1] \"SampleData1/Fold_1\" \"train_data\"         \"train_data.QC\"     \n",
      "[4] \"1\"                 \n",
      "[1] \"SampleData1\" \"Fold_1\"     \n",
      "[1] \"SampleData1\"\n",
      "[1] 14\n",
      "[1] 14\n",
      "[1] 14\n",
      "Rscript Module1.R SampleData1/Fold_1  train_data train_data.QC 1\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_1/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_1/train_data\n",
      "  --extract SampleData1/Fold_1/train_data.QC.snplist\n",
      "  --out SampleData1/Fold_1/train_data.QC\n",
      "  --rel-cutoff 0.125\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (178 males, 202 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 543113 variants remaining.\n",
      "Using up to 8 threads (change this with --threads).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999923.\n",
      "543113 variants and 380 people pass filters and QC (before --rel-cutoff).\n",
      "Phenotype data is quantitative.\n",
      "Excluding 21313 variants on non-autosomes from relationship matrix calc.\n",
      "Relationship matrix calculation complete.\n",
      "0 people excluded by --rel-cutoff.\n",
      "Remaining sample IDs written to SampleData1/Fold_1/train_data.QC.rel.id .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_1/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --a1-allele SampleData1/Fold_1/train_data.a1\n",
      "  --bfile SampleData1/Fold_1/train_data\n",
      "  --exclude SampleData1/Fold_1/train_data.mismatch\n",
      "  --extract SampleData1/Fold_1/train_data.QC.snplist\n",
      "  --keep SampleData1/Fold_1/train_data.QC.rel.id\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_1/train_data.QC\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (178 males, 202 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 543113 variants remaining.\n",
      "--exclude: 492382 variants remaining.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99992.\n",
      "--a1-allele: 492382 assignments made.\n",
      "492382 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_1/train_data.QC.bed +\n",
      "SampleData1/Fold_1/train_data.QC.bim + SampleData1/Fold_1/train_data.QC.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_2/train_data.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/SampleData1_QC\n",
      "  --keep SampleData1/Fold_2/train_data.fam\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_2/train_data\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "475 people (227 males, 248 females) loaded from .fam.\n",
      "475 phenotype values loaded from .fam.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999896.\n",
      "551892 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_2/train_data.bed +\n",
      "SampleData1/Fold_2/train_data.bim + SampleData1/Fold_2/train_data.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_2/test_data.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/SampleData1_QC\n",
      "  --keep SampleData1/Fold_2/test_data.fam\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_2/test_data\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "475 people (227 males, 248 females) loaded from .fam.\n",
      "475 phenotype values loaded from .fam.\n",
      "--keep: 95 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 95 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999896.\n",
      "551892 variants and 95 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_2/test_data.bed +\n",
      "SampleData1/Fold_2/test_data.bim + SampleData1/Fold_2/test_data.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_2/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_2/train_data\n",
      "  --geno 0.1\n",
      "  --hwe 1e-6\n",
      "  --maf 0.01\n",
      "  --make-just-fam\n",
      "  --mind 0.1\n",
      "  --out SampleData1/Fold_2/train_data.QC\n",
      "  --write-snplist\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (180 males, 200 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "0 people removed due to missing genotype data (--mind).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999896.\n",
      "1 variant removed due to missing genotype data (--geno).\n",
      "--hwe: 847 variants removed due to Hardy-Weinberg exact test.\n",
      "8536 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "542508 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "List of variant IDs written to SampleData1/Fold_2/train_data.QC.snplist .\n",
      "--make-just-fam to SampleData1/Fold_2/train_data.QC.fam ... done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_2/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_2/train_data\n",
      "  --extract SampleData1/Fold_2/train_data.QC.snplist\n",
      "  --het\n",
      "  --keep SampleData1/Fold_2/train_data.QC.fam\n",
      "  --out SampleData1/Fold_2/train_data.QC\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (180 males, 200 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 542508 variants remaining.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 1011121314151617181920212223242526272829303132333435363738394041424344454647484950515253545556575859606162636465666768697071727374757677787980818283848586%"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --hwe observation counts vary by more than 10%, due to the X\n",
      "chromosome.  You may want to use a less stringent --hwe p-value threshold for X\n",
      "chromosome variants.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\b8788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999898.\n",
      "542508 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--het: 521405 variants scanned, report written to\n",
      "SampleData1/Fold_2/train_data.QC.het .\n",
      "[1] \"SampleData1/Fold_2\" \"train_data\"         \"train_data.QC\"     \n",
      "[4] \"1\"                 \n",
      "[1] \"SampleData1\" \"Fold_2\"     \n",
      "[1] \"SampleData1\"\n",
      "[1] 14\n",
      "[1] 14\n",
      "[1] 14\n",
      "Rscript Module1.R SampleData1/Fold_2  train_data train_data.QC 1\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_2/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_2/train_data\n",
      "  --extract SampleData1/Fold_2/train_data.QC.snplist\n",
      "  --out SampleData1/Fold_2/train_data.QC\n",
      "  --rel-cutoff 0.125\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (180 males, 200 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 542508 variants remaining.\n",
      "Using up to 8 threads (change this with --threads).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999898.\n",
      "542508 variants and 380 people pass filters and QC (before --rel-cutoff).\n",
      "Phenotype data is quantitative.\n",
      "Excluding 21103 variants on non-autosomes from relationship matrix calc.\n",
      "Relationship matrix calculation complete.\n",
      "0 people excluded by --rel-cutoff.\n",
      "Remaining sample IDs written to SampleData1/Fold_2/train_data.QC.rel.id .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_2/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --a1-allele SampleData1/Fold_2/train_data.a1\n",
      "  --bfile SampleData1/Fold_2/train_data\n",
      "  --exclude SampleData1/Fold_2/train_data.mismatch\n",
      "  --extract SampleData1/Fold_2/train_data.QC.snplist\n",
      "  --keep SampleData1/Fold_2/train_data.QC.rel.id\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_2/train_data.QC\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (180 males, 200 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 542508 variants remaining.\n",
      "--exclude: 491965 variants remaining.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999894.\n",
      "--a1-allele: 491965 assignments made.\n",
      "491965 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_2/train_data.QC.bed +\n",
      "SampleData1/Fold_2/train_data.QC.bim + SampleData1/Fold_2/train_data.QC.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_3/train_data.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/SampleData1_QC\n",
      "  --keep SampleData1/Fold_3/train_data.fam\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_3/train_data\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "475 people (227 males, 248 females) loaded from .fam.\n",
      "475 phenotype values loaded from .fam.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999871.\n",
      "551892 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_3/train_data.bed +\n",
      "SampleData1/Fold_3/train_data.bim + SampleData1/Fold_3/train_data.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_3/test_data.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/SampleData1_QC\n",
      "  --keep SampleData1/Fold_3/test_data.fam\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_3/test_data\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "475 people (227 males, 248 females) loaded from .fam.\n",
      "475 phenotype values loaded from .fam.\n",
      "--keep: 95 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 95 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999998.\n",
      "551892 variants and 95 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_3/test_data.bed +\n",
      "SampleData1/Fold_3/test_data.bim + SampleData1/Fold_3/test_data.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_3/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_3/train_data\n",
      "  --geno 0.1\n",
      "  --hwe 1e-6\n",
      "  --maf 0.01\n",
      "  --make-just-fam\n",
      "  --mind 0.1\n",
      "  --out SampleData1/Fold_3/train_data.QC\n",
      "  --write-snplist\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (181 males, 199 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "0 people removed due to missing genotype data (--mind).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999871.\n",
      "1 variant removed due to missing genotype data (--geno).\n",
      "--hwe: 841 variants removed due to Hardy-Weinberg exact test.\n",
      "8339 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "542711 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "List of variant IDs written to SampleData1/Fold_3/train_data.QC.snplist .\n",
      "--make-just-fam to SampleData1/Fold_3/train_data.QC.fam ... done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_3/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_3/train_data\n",
      "  --extract SampleData1/Fold_3/train_data.QC.snplist\n",
      "  --het\n",
      "  --keep SampleData1/Fold_3/train_data.QC.fam\n",
      "  --out SampleData1/Fold_3/train_data.QC\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (181 males, 199 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 542711 variants remaining.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182%"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --hwe observation counts vary by more than 10%, due to the X\n",
      "chromosome.  You may want to use a less stringent --hwe p-value threshold for X\n",
      "chromosome variants.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\b838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999872.\n",
      "542711 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--het: 521511 variants scanned, report written to\n",
      "SampleData1/Fold_3/train_data.QC.het .\n",
      "[1] \"SampleData1/Fold_3\" \"train_data\"         \"train_data.QC\"     \n",
      "[4] \"1\"                 \n",
      "[1] \"SampleData1\" \"Fold_3\"     \n",
      "[1] \"SampleData1\"\n",
      "[1] 14\n",
      "[1] 14\n",
      "[1] 14\n",
      "Rscript Module1.R SampleData1/Fold_3  train_data train_data.QC 1\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_3/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_3/train_data\n",
      "  --extract SampleData1/Fold_3/train_data.QC.snplist\n",
      "  --out SampleData1/Fold_3/train_data.QC\n",
      "  --rel-cutoff 0.125\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (181 males, 199 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 542711 variants remaining.\n",
      "Using up to 8 threads (change this with --threads).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999872.\n",
      "542711 variants and 380 people pass filters and QC (before --rel-cutoff).\n",
      "Phenotype data is quantitative.\n",
      "Excluding 21200 variants on non-autosomes from relationship matrix calc.\n",
      "Relationship matrix calculation complete.\n",
      "0 people excluded by --rel-cutoff.\n",
      "Remaining sample IDs written to SampleData1/Fold_3/train_data.QC.rel.id .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_3/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --a1-allele SampleData1/Fold_3/train_data.a1\n",
      "  --bfile SampleData1/Fold_3/train_data\n",
      "  --exclude SampleData1/Fold_3/train_data.mismatch\n",
      "  --extract SampleData1/Fold_3/train_data.QC.snplist\n",
      "  --keep SampleData1/Fold_3/train_data.QC.rel.id\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_3/train_data.QC\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (181 males, 199 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 542711 variants remaining.\n",
      "--exclude: 492072 variants remaining.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999868.\n",
      "--a1-allele: 492072 assignments made.\n",
      "492072 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_3/train_data.QC.bed +\n",
      "SampleData1/Fold_3/train_data.QC.bim + SampleData1/Fold_3/train_data.QC.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_4/train_data.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/SampleData1_QC\n",
      "  --keep SampleData1/Fold_4/train_data.fam\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_4/train_data\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "475 people (227 males, 248 females) loaded from .fam.\n",
      "475 phenotype values loaded from .fam.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999896.\n",
      "551892 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_4/train_data.bed +\n",
      "SampleData1/Fold_4/train_data.bim + SampleData1/Fold_4/train_data.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_4/test_data.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/SampleData1_QC\n",
      "  --keep SampleData1/Fold_4/test_data.fam\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_4/test_data\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "475 people (227 males, 248 females) loaded from .fam.\n",
      "475 phenotype values loaded from .fam.\n",
      "--keep: 95 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 95 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999896.\n",
      "551892 variants and 95 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_4/test_data.bed +\n",
      "SampleData1/Fold_4/test_data.bim + SampleData1/Fold_4/test_data.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_4/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_4/train_data\n",
      "  --geno 0.1\n",
      "  --hwe 1e-6\n",
      "  --maf 0.01\n",
      "  --make-just-fam\n",
      "  --mind 0.1\n",
      "  --out SampleData1/Fold_4/train_data.QC\n",
      "  --write-snplist\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (186 males, 194 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "0 people removed due to missing genotype data (--mind).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999896.\n",
      "1 variant removed due to missing genotype data (--geno).\n",
      "--hwe: 828 variants removed due to Hardy-Weinberg exact test.\n",
      "8533 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "542530 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "List of variant IDs written to SampleData1/Fold_4/train_data.QC.snplist .\n",
      "--make-just-fam to SampleData1/Fold_4/train_data.QC.fam ... done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_4/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_4/train_data\n",
      "  --extract SampleData1/Fold_4/train_data.QC.snplist\n",
      "  --het\n",
      "  --keep SampleData1/Fold_4/train_data.QC.fam\n",
      "  --out SampleData1/Fold_4/train_data.QC\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (186 males, 194 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 542530 variants remaining.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596%"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --hwe observation counts vary by more than 10%, due to the X\n",
      "chromosome.  You may want to use a less stringent --hwe p-value threshold for X\n",
      "chromosome variants.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\b97989 done.\n",
      "Total genotyping rate is 0.999898.\n",
      "542530 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--het: 521557 variants scanned, report written to\n",
      "SampleData1/Fold_4/train_data.QC.het .\n",
      "[1] \"SampleData1/Fold_4\" \"train_data\"         \"train_data.QC\"     \n",
      "[4] \"1\"                 \n",
      "[1] \"SampleData1\" \"Fold_4\"     \n",
      "[1] \"SampleData1\"\n",
      "[1] 14\n",
      "[1] 14\n",
      "[1] 14\n",
      "Rscript Module1.R SampleData1/Fold_4  train_data train_data.QC 1\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_4/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --bfile SampleData1/Fold_4/train_data\n",
      "  --extract SampleData1/Fold_4/train_data.QC.snplist\n",
      "  --out SampleData1/Fold_4/train_data.QC\n",
      "  --rel-cutoff 0.125\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (186 males, 194 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 542530 variants remaining.\n",
      "Using up to 8 threads (change this with --threads).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999898.\n",
      "542530 variants and 380 people pass filters and QC (before --rel-cutoff).\n",
      "Phenotype data is quantitative.\n",
      "Excluding 20973 variants on non-autosomes from relationship matrix calc.\n",
      "Relationship matrix calculation complete.\n",
      "0 people excluded by --rel-cutoff.\n",
      "Remaining sample IDs written to SampleData1/Fold_4/train_data.QC.rel.id .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SampleData1/Fold_4/train_data.QC.log.\n",
      "Options in effect:\n",
      "  --a1-allele SampleData1/Fold_4/train_data.a1\n",
      "  --bfile SampleData1/Fold_4/train_data\n",
      "  --exclude SampleData1/Fold_4/train_data.mismatch\n",
      "  --extract SampleData1/Fold_4/train_data.QC.snplist\n",
      "  --keep SampleData1/Fold_4/train_data.QC.rel.id\n",
      "  --make-bed\n",
      "  --out SampleData1/Fold_4/train_data.QC\n",
      "\n",
      "63761 MB RAM detected; reserving 31880 MB for main workspace.\n",
      "551892 variants loaded from .bim file.\n",
      "380 people (186 males, 194 females) loaded from .fam.\n",
      "380 phenotype values loaded from .fam.\n",
      "--extract: 542530 variants remaining.\n",
      "--exclude: 492135 variants remaining.\n",
      "--keep: 380 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 380 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999894.\n",
      "--a1-allele: 492135 assignments made.\n",
      "492135 variants and 380 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "--make-bed to SampleData1/Fold_4/train_data.QC.bed +\n",
      "SampleData1/Fold_4/train_data.QC.bim + SampleData1/Fold_4/train_data.QC.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "import os\n",
    "import pandas as pd\n",
    "from sklearn.model_selection import StratifiedKFold\n",
    "import subprocess\n",
    "from sklearn.model_selection import KFold, cross_val_score\n",
    "\n",
    "# Step 1: Read the Fam file\n",
    "\n",
    "input_file_path = filedirec+os.sep+newfilename+'.fam'\n",
    "df = pd.read_csv(input_file_path,sep=\"\\s+\",header=None)\n",
    "\n",
    "# Step 2: Create 5 directories for storing fold information\n",
    "output_directory_base = filedirec\n",
    "os.makedirs(output_directory_base, exist_ok=True)\n",
    "\n",
    "# Step 3: Split the data into 5 folds using cross validation \n",
    "fold_column = 5  # fifth column contains phenotypes\n",
    "kf = KFold(n_splits=5, shuffle=True, random_state=42)\n",
    "phenotype_col = 5\n",
    "\n",
    "# The following code is for binary phenotype.\n",
    "skf = StratifiedKFold(n_splits=5, shuffle=True, random_state=42)\n",
    "column_values = df[phenotype_col].unique()\n",
    "\n",
    "if set(column_values) == {1, 2}:\n",
    "    print(\"The column contains only 0 and 1.\")\n",
    "    #exit(0)\n",
    "    for fold_id, (train_index, test_index) in enumerate(skf.split(df, df[phenotype_col])):\n",
    "        fold_directory = os.path.join(output_directory_base, f'Fold_{fold_id}')\n",
    "\n",
    "        train_file_name = \"train_data\"\n",
    "        test_file_name = \"test_data\"\n",
    "\n",
    "        new_train_file_name = \"train_data.QC\"\n",
    "        new_test_file_name = \"test_data.QC\"\n",
    "\n",
    "        os.makedirs(fold_directory, exist_ok=True)\n",
    "        #\"\"\"\n",
    "        # Save train and test data to separate CSV files\n",
    "        train_data = df.iloc[train_index]\n",
    "        test_data = df.iloc[test_index]\n",
    "        \n",
    "        train_data.to_csv(os.path.join(fold_directory, 'train_data.fam'),sep=\"\\t\",header=False,index=False)\n",
    "        test_data.to_csv(os.path.join(fold_directory, 'test_data.fam'),sep=\"\\t\",header=False,index=False)\n",
    "        #print(train_data)\n",
    "        #exit(0)\n",
    "        \n",
    "        #exit(0)\n",
    "        # Step 4: Use PLINK to extract test and train samples for each fold\n",
    "        plink_train_command = [\n",
    "            './plink',\n",
    "            '--bfile', filedirec+os.sep+newfilename,\n",
    "            '--keep', os.path.join(fold_directory, train_file_name+'.fam'),\n",
    "            '--make-bed',\n",
    "            '--out', os.path.join(fold_directory, train_file_name)\n",
    "        ]\n",
    "\n",
    "        plink_test_command = [\n",
    "            './plink',\n",
    "            '--bfile', filedirec+os.sep+newfilename,\n",
    "            '--keep', os.path.join(fold_directory, test_file_name+'.fam'),\n",
    "            '--make-bed',\n",
    "            '--out', os.path.join(fold_directory, test_file_name)\n",
    "        ]\n",
    "\n",
    "        subprocess.run(plink_train_command)\n",
    "        subprocess.run(plink_test_command)\n",
    "        \n",
    "        covfile = filedirec+os.sep+newfilename+'.cov'\n",
    "        covfile = pd.read_csv(covfile)\n",
    "\n",
    "        cov_train_data = covfile.iloc[train_index]\n",
    "        cov_test_data = covfile.iloc[test_index]\n",
    "\n",
    "        cov_train_data.to_csv(os.path.join(fold_directory, train_file_name+'.cov'),sep=\",\",index=False)\n",
    "        cov_test_data.to_csv(os.path.join(fold_directory, test_file_name+'.cov'),sep=\",\",index=False)\n",
    "\n",
    "        #exit(0)\n",
    "        ### perform Quality controls on the training data only.\n",
    "        plink_command_1 = [\n",
    "            './plink',\n",
    "            '--bfile', os.path.join(fold_directory, train_file_name),\n",
    "            '--maf', '0.01',\n",
    "            '--hwe', '1e-6',\n",
    "            '--geno', '0.1',\n",
    "            '--mind', '0.1',\n",
    "            '--write-snplist',\n",
    "            '--make-just-fam',\n",
    "            '--out', os.path.join(fold_directory, new_train_file_name)\n",
    "        ]\n",
    "    \n",
    "        subprocess.run(plink_command_1)\n",
    "        \n",
    "        # Command 2\n",
    "        # Perform pruning. Skip it.\n",
    "        plink_command_2 = [\n",
    "            './plink',\n",
    "            '--bfile', os.path.join(fold_directory, train_file_name),\n",
    "            '--keep', os.path.join(fold_directory, new_train_file_name+'.fam'),\n",
    "            '--extract', os.path.join(fold_directory, new_train_file_name+'.snplist'),\n",
    "            '--indep-pairwise', '200', '50', '0.25',\n",
    "            '--out', os.path.join(fold_directory, new_train_file_name)\n",
    "        ]\n",
    "    \n",
    "        #subprocess.run(plink_command_2)\n",
    "        \n",
    "        # Command 3\n",
    "        plink_command_3 = [\n",
    "            './plink',\n",
    "            '--bfile', os.path.join(fold_directory, train_file_name),\n",
    "            #'--extract', os.path.join(fold_directory, new_train_file_name+'.prune.in'),\n",
    "            '--extract', os.path.join(fold_directory, new_train_file_name+'.snplist'),\n",
    "            '--keep', os.path.join(fold_directory, new_train_file_name+'.fam'),\n",
    "            '--het',\n",
    "            '--out', os.path.join(fold_directory, new_train_file_name)\n",
    "        ]\n",
    "    \n",
    "        subprocess.run(plink_command_3)\n",
    "        # Invoked R functions.\n",
    "        \n",
    "        os.system(\"Rscript Module1.R \"+os.path.join(fold_directory)+\"  \"+train_file_name+\" \"+new_train_file_name+ \" \"+\"1\")\n",
    "        print(\"Rscript Module1.R \"+os.path.join(fold_directory)+\"  \"+train_file_name+\" \"+new_train_file_name+ \" \"+\"1\")\n",
    "         \n",
    "        \n",
    "        # Code for sex check: The sample data have 22 chromosomes, so this operation is skipped.\n",
    "        # If the Chromosome X is not available, then do not execute the following two commands. \n",
    "        plink_command = [\n",
    "        './plink',\n",
    "        '--bfile', os.path.join(fold_directory, 'train_data'),\n",
    "        #'--extract', os.path.join(fold_directory, 'train_data.QC.prune.in'),\n",
    "        '--extract', os.path.join(fold_directory, new_train_file_name+'.snplist'),\n",
    "        '--keep', os.path.join(fold_directory, 'train_data.valid.sample'),\n",
    "        '--check-sex',\n",
    "        '--out', os.path.join(fold_directory, 'train_data.QC')\n",
    "        ]\n",
    "        \n",
    "        # Invoke the PLINK command using subprocess for sex check\n",
    "        #subprocess.run(plink_command)\n",
    "        #os.system(\"Rscript Module1.R \"+os.path.join(fold_directory)+\"  \"+train_file_name+\" \"+new_train_file_name+ \" \"+\"2\")\n",
    "        \n",
    "        #\"\"\"\n",
    "    \n",
    "        plink_command_1 = [\n",
    "        './plink',\n",
    "        '--bfile', os.path.join(fold_directory, train_file_name),\n",
    "        #'--extract', os.path.join(fold_directory, new_train_file_name+'.prune.in'),\n",
    "        # If you use sex check, then use the following line. Otherwise, it is not required.\n",
    "        '--extract', os.path.join(fold_directory, new_train_file_name+'.snplist'),\n",
    "\n",
    "        # Uncomment the following line if the sex Chromosome is available.\n",
    "        # Kindly not, if you perform Sex check, and some people are removed, you have to\n",
    "        # remove people from covariate file and Phenotype file.\n",
    "            \n",
    "        #'--keep', os.path.join(fold_directory, 'train_data.QC.valid'),\n",
    "        '--rel-cutoff', '0.125',\n",
    "        '--out', os.path.join(fold_directory, new_train_file_name)\n",
    "        ]\n",
    "        \n",
    "        subprocess.run(plink_command_1)\n",
    "         \n",
    "         \n",
    "        plink_command_2 = [\n",
    "            './plink',\n",
    "            '--bfile', os.path.join(fold_directory, train_file_name),\n",
    "            '--make-bed',\n",
    "            '--keep', os.path.join(fold_directory, new_train_file_name+'.rel.id'),\n",
    "            '--out', os.path.join(fold_directory, new_train_file_name),\n",
    "            '--extract', os.path.join(fold_directory, new_train_file_name+'.snplist'),\n",
    "            '--exclude', os.path.join(fold_directory, train_file_name+'.mismatch'),\n",
    "            '--a1-allele', os.path.join(fold_directory, train_file_name+'.a1')\n",
    "        ]\n",
    "    \n",
    "        subprocess.run(plink_command_2)\n",
    "         \n",
    "else:\n",
    "    print(\"The column does not contain only 0 and 1.\")\n",
    "    for fold_id, (train_index, test_index) in enumerate(kf.split(df, df[phenotype_col])):\n",
    "        fold_directory = os.path.join(output_directory_base, f'Fold_{fold_id}')\n",
    "\n",
    "        train_file_name = \"train_data\"\n",
    "        test_file_name = \"test_data\"\n",
    "\n",
    "        new_train_file_name = \"train_data.QC\"\n",
    "        new_test_file_name = \"test_data.QC\"\n",
    "\n",
    "        os.makedirs(fold_directory, exist_ok=True)\n",
    "        #\"\"\"\n",
    "        # Save train and test data to separate CSV files\n",
    "        train_data = df.iloc[train_index]\n",
    "        test_data = df.iloc[test_index]\n",
    "        \n",
    "        train_data.to_csv(os.path.join(fold_directory, 'train_data.fam'),sep=\"\\t\",header=False,index=False)\n",
    "        test_data.to_csv(os.path.join(fold_directory, 'test_data.fam'),sep=\"\\t\",header=False,index=False)\n",
    "        #print(train_data)\n",
    "        #exit(0)\n",
    "        \n",
    "        #exit(0)\n",
    "        # Step 4: Use PLINK to extract test and train samples for each fold\n",
    "        plink_train_command = [\n",
    "            './plink',\n",
    "            '--bfile', filedirec+os.sep+newfilename,\n",
    "            '--keep', os.path.join(fold_directory, train_file_name+'.fam'),\n",
    "            '--make-bed',\n",
    "            '--out', os.path.join(fold_directory, train_file_name)\n",
    "        ]\n",
    "\n",
    "        plink_test_command = [\n",
    "            './plink',\n",
    "            '--bfile', filedirec+os.sep+newfilename,\n",
    "            '--keep', os.path.join(fold_directory, test_file_name+'.fam'),\n",
    "            '--make-bed',\n",
    "            '--out', os.path.join(fold_directory, test_file_name)\n",
    "        ]\n",
    "\n",
    "        subprocess.run(plink_train_command)\n",
    "        subprocess.run(plink_test_command)\n",
    "        \n",
    "        covfile = filedirec+os.sep+newfilename+'.cov'\n",
    "        covfile = pd.read_csv(covfile)\n",
    "\n",
    "        cov_train_data = covfile.iloc[train_index]\n",
    "        cov_test_data = covfile.iloc[test_index]\n",
    "\n",
    "        cov_train_data.to_csv(os.path.join(fold_directory, train_file_name+'.cov'),sep=\",\",index=False)\n",
    "        cov_test_data.to_csv(os.path.join(fold_directory, test_file_name+'.cov'),sep=\",\",index=False)\n",
    "\n",
    "        #exit(0)\n",
    "        ### perform Quality controls on the training data only.\n",
    "        plink_command_1 = [\n",
    "            './plink',\n",
    "            '--bfile', os.path.join(fold_directory, train_file_name),\n",
    "            '--maf', '0.01',\n",
    "            '--hwe', '1e-6',\n",
    "            '--geno', '0.1',\n",
    "            '--mind', '0.1',\n",
    "            '--write-snplist',\n",
    "            '--make-just-fam',\n",
    "            '--out', os.path.join(fold_directory, new_train_file_name)\n",
    "        ]\n",
    "    \n",
    "        subprocess.run(plink_command_1)\n",
    "        \n",
    "        # Command 2\n",
    "        plink_command_2 = [\n",
    "            './plink',\n",
    "            '--bfile', os.path.join(fold_directory, train_file_name),\n",
    "            '--keep', os.path.join(fold_directory, new_train_file_name+'.fam'),\n",
    "            '--extract', os.path.join(fold_directory, new_train_file_name+'.snplist'),\n",
    "            '--indep-pairwise', '200', '50', '0.25',\n",
    "            '--out', os.path.join(fold_directory, new_train_file_name)\n",
    "        ]\n",
    "    \n",
    "        #subprocess.run(plink_command_2)\n",
    "        \n",
    "        # Command 3\n",
    "        plink_command_3 = [\n",
    "            './plink',\n",
    "            '--bfile', os.path.join(fold_directory, train_file_name),\n",
    "            #'--extract', os.path.join(fold_directory, new_train_file_name+'.prune.in'),\n",
    "            '--extract', os.path.join(fold_directory, new_train_file_name+'.snplist'),\n",
    "            '--keep', os.path.join(fold_directory, new_train_file_name+'.fam'),\n",
    "            '--het',\n",
    "            '--out', os.path.join(fold_directory, new_train_file_name)\n",
    "        ]\n",
    "    \n",
    "        subprocess.run(plink_command_3)\n",
    "        # Invoked R functions.\n",
    "        \n",
    "        os.system(\"Rscript Module1.R \"+os.path.join(fold_directory)+\"  \"+train_file_name+\" \"+new_train_file_name+ \" \"+\"1\")\n",
    "        print(\"Rscript Module1.R \"+os.path.join(fold_directory)+\"  \"+train_file_name+\" \"+new_train_file_name+ \" \"+\"1\")\n",
    "        \n",
    "        \n",
    "        \n",
    "        # Code for sex check: The sample data have 22 chromosomes, so this operation is skipped.\n",
    "        # If the Chromosome X is not available, then do not execute the following two commands. \n",
    "        #\"\"\"\n",
    "        plink_command = [\n",
    "        './plink',\n",
    "        '--bfile', os.path.join(fold_directory, 'train_data'),\n",
    "        #'--extract', os.path.join(fold_directory, 'train_data.QC.prune.in'),\n",
    "        '--extract', os.path.join(fold_directory, new_train_file_name+'.snplist'),\n",
    "        '--keep', os.path.join(fold_directory, 'train_data.valid.sample'),\n",
    "        '--check-sex',\n",
    "        '--out', os.path.join(fold_directory, 'train_data.QC')\n",
    "        ]\n",
    "        \n",
    "        # Invoke the PLINK command using subprocess for sex check\n",
    "         \n",
    "        #subprocess.run(plink_command)\n",
    "        #os.system(\"Rscript Module1.R \"+os.path.join(fold_directory)+\"  \"+train_file_name+\" \"+new_train_file_name+ \" \"+\"2\")\n",
    "        \n",
    "        #\"\"\"\n",
    "    \n",
    "        plink_command_1 = [\n",
    "        './plink',\n",
    "        '--bfile', os.path.join(fold_directory, train_file_name),\n",
    "        #'--extract', os.path.join(fold_directory, new_train_file_name+'.prune.in'),\n",
    "        # If you use sex check then use the following line. Otherwise it is not required.\n",
    "        '--extract', os.path.join(fold_directory, new_train_file_name+'.snplist'),\n",
    "        #'--keep', os.path.join(fold_directory, 'train_data.QC.valid'),\n",
    "        '--rel-cutoff', '0.125',\n",
    "        '--out', os.path.join(fold_directory, new_train_file_name)\n",
    "        ]\n",
    "        \n",
    "        subprocess.run(plink_command_1)\n",
    "        #exit(0)\n",
    "        \n",
    "        #exit(0)\n",
    "        # Command 2\n",
    "        plink_command_2 = [\n",
    "            './plink',\n",
    "            '--bfile', os.path.join(fold_directory, train_file_name),\n",
    "            '--make-bed',\n",
    "            '--keep', os.path.join(fold_directory, new_train_file_name+'.rel.id'),\n",
    "            '--out', os.path.join(fold_directory, new_train_file_name),\n",
    "            '--extract', os.path.join(fold_directory, new_train_file_name+'.snplist'),\n",
    "            '--exclude', os.path.join(fold_directory, train_file_name+'.mismatch'),\n",
    "            '--a1-allele', os.path.join(fold_directory, train_file_name+'.a1')\n",
    "        ]\n",
    "    \n",
    "        subprocess.run(plink_command_2)\n",
    "        #exit(0)\n",
    " \n",
    "\n",
    "\n",
    "\n",
    " \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "749d05bd",
   "metadata": {},
   "source": [
    "## Check final files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "61439ac4-f4cc-4b66-946b-12773a4a9a60",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File Name            Fold 0 Fold 1 Fold 2 Fold 3 Fold 4\n",
      "train_data.QC.bed    yes   yes   yes   yes   yes  \n",
      "train_data.QC.bim    yes   yes   yes   yes   yes  \n",
      "train_data.QC.fam    yes   yes   yes   yes   yes  \n",
      "train_data.cov       yes   yes   yes   yes   yes  \n",
      "test_data.bed        yes   yes   yes   yes   yes  \n",
      "test_data.bim        yes   yes   yes   yes   yes  \n",
      "test_data.fam        yes   yes   yes   yes   yes  \n",
      "test_data.cov        yes   yes   yes   yes   yes  \n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# List of file names to check for existence\n",
    "files = [\n",
    "    \"train_data.QC.bed\",\n",
    "    \"train_data.QC.bim\",\n",
    "    \"train_data.QC.fam\",\n",
    "    \"train_data.cov\",\n",
    "    \"test_data.bed\",\n",
    "    \"test_data.bim\",\n",
    "    \"test_data.fam\",\n",
    "    \"test_data.cov\"\n",
    "]\n",
    "\n",
    "# Directory where the files are expected to be found\n",
    "\n",
    "# Print the table header\n",
    "print(\"{:<20} {:<5} {:<5} {:<5} {:<5} {:<5}\".format(\"File Name\", \"Fold 0\", \"Fold 1\", \"Fold 2\", \"Fold 3\", \"Fold 4\"))\n",
    "\n",
    "# Loop through each file name in the list\n",
    "for file in files:\n",
    "    # Create a list to store the existence status for each fold\n",
    "    status = []\n",
    "    # Check for each fold from 0 to 4\n",
    "    for fold_number in range(5):\n",
    "        # Check if the file exists in the specified directory for the given fold\n",
    "        #print(os.path.join(\"./\",filedirec, f\"Fold_{fold_number}\", file))\n",
    "        if os.path.exists(filedirec + os.sep + \"Fold_\" + str(fold_number) + os.sep + file):\n",
    "            status.append(\"yes\")\n",
    "        else:\n",
    "            status.append(\"no\")\n",
    "    \n",
    "    # Print the file name and its status for each fold\n",
    "    print(\"{:<20} {:<5} {:<5} {:<5} {:<5} {:<5}\".format(file, *status))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e8c8728a-5179-474c-a93d-d52421fab5bb",
   "metadata": {},
   "source": [
    "We will have the following directories if everything works fine.\n",
    "\n",
    "```\n",
    "├── Fold_0\n",
    "│   ├── test_data.bed\n",
    "│   ├── test_data.bim\n",
    "│   ├── test_data.cov\n",
    "│   ├── test_data.fam\n",
    "│   ├── test_data.log\n",
    "│   ├── train_data.a1\n",
    "│   ├── train_data.bed\n",
    "│   ├── train_data.bim\n",
    "│   ├── train_data.cov\n",
    "│   ├── train_data.fam\n",
    "│   ├── train_data.log\n",
    "│   ├── train_data.mismatch\n",
    "│   ├── train_data.QC.bed\n",
    "│   ├── train_data.QC.bim\n",
    "│   ├── train_data.QC.fam\n",
    "│   ├── train_data.QC.het\n",
    "│   ├── train_data.QC.log\n",
    "│   ├── train_data.QC.rel.id\n",
    "│   ├── train_data.QC.sexcheck\n",
    "│   ├── train_data.QC.snplist\n",
    "│   ├── train_data.QC.valid\n",
    "│   └── train_data.valid.sample\n",
    "├── Fold_1\n",
    " \n",
    "├── Fold_2\n",
    " \n",
    "├── Fold_3\n",
    " \n",
    "├── Fold_4\n",
    "\n",
    "├── PeopleWithPhenotype.txt\n",
    "├── SampleData1.bed\n",
    "├── SampleData1.bim\n",
    "├── SampleData1.cov\n",
    "├── SampleData1.fam\n",
    "├── SampleData1.gz\n",
    "├── SampleData1.height\n",
    "├── SampleData1_QC.bed\n",
    "├── SampleData1_QC.bim\n",
    "├── SampleData1_QC.cov\n",
    "├── SampleData1_QC.fam\n",
    "└── SampleData1_QC.log\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "84dd5850",
   "metadata": {},
   "source": [
    "## Important Note\n",
    "\n",
    "1. Kindly ensure you have all the files required for the next step after the completion of this step.\n",
    "2. It is better to pass the dataset on which quality controls have already been performed.\n",
    "3. We considered genotype files for which chromosomes 1 to 22 are available, and sex information is present. If you want to use the function in the script to check the sex information, ensure the genotype file includes other chromosomes as well.\n",
    "4. Go through the logs generated by the code if an error occurs. Even if no error occurs, it is always good to ensure that the log file does not produce any errors.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21acd715",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}