<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Introduction - PRSTools &mdash; Polygenic Risk Score Tools 1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=f2a433a1"></script>
        <script src="_static/doctools.js?v=888ff710"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Step1-DownloadSampleDataandSetupEnvironment" href="Step1-DownloadSampleDataandSetupEnvironment.html" />
    <link rel="prev" title="Welcome to Polygenic Risk Score Tools’s documentation!" href="index.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            Polygenic Risk Score Tools
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Introduction - PRSTools</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#background">Background</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#key-challenges">Key Challenges</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#repository-overview">Repository Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="#conclusion">Conclusion</a></li>
<li class="toctree-l2"><a class="reference internal" href="#prs-tools-included-in-the-analysis">PRS Tools Included in the Analysis.</a></li>
<li class="toctree-l2"><a class="reference internal" href="#conda-environment">Conda Environment</a></li>
<li class="toctree-l2"><a class="reference internal" href="#results">Results</a></li>
<li class="toctree-l2"><a class="reference internal" href="#related-projects">Related Projects.</a></li>
<li class="toctree-l2"><a class="reference internal" href="#author-information">Author Information.</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Step1-DownloadSampleDataandSetupEnvironment.html">Step1-DownloadSampleDataandSetupEnvironment</a></li>
<li class="toctree-l1"><a class="reference internal" href="Step2-GWASAndIndividualGenotypeDataQualityControls.html">Step2-GWASAndIndividualGenotypeDataQualityControls</a></li>
<li class="toctree-l1"><a class="reference internal" href="NotebooksWorkflow.html">Polygenic Risk Score Tools Notebooks Workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="Plink.html">Plink</a></li>
<li class="toctree-l1"><a class="reference internal" href="PRSice-2.html">PRSice-2</a></li>
<li class="toctree-l1"><a class="reference internal" href="DBSLMM.html">DBSLMM</a></li>
<li class="toctree-l1"><a class="reference internal" href="GCTA.html">GCTA</a></li>
<li class="toctree-l1"><a class="reference internal" href="Lassosum.html">Lassosum</a></li>
<li class="toctree-l1"><a class="reference internal" href="LDpred-2-Inf.html">LDpred-2-Inf</a></li>
<li class="toctree-l1"><a class="reference internal" href="LDpred-2-Grid.html">LDpred-2-Grid</a></li>
<li class="toctree-l1"><a class="reference internal" href="LDpred-2-Auto.html">LDpred-2-Auto</a></li>
<li class="toctree-l1"><a class="reference internal" href="LDpred-2-Lassosum.html">LDpred-2-Lassosum</a></li>
<li class="toctree-l1"><a class="reference internal" href="LDpred-funct-code.html">LDpred-funct</a></li>
<li class="toctree-l1"><a class="reference internal" href="GCTB-SBayesR.html">GCTB: Calculating PRS with SBayesR</a></li>
<li class="toctree-l1"><a class="reference internal" href="GCTB-SBayesRC.html">GCTB-SBayesRC</a></li>
<li class="toctree-l1"><a class="reference internal" href="LDAK-GenotypeData.html">LDAK-GenotypeData</a></li>
<li class="toctree-l1"><a class="reference internal" href="LDAK-GWAS.html">LDAK-GWAS</a></li>
<li class="toctree-l1"><a class="reference internal" href="PRScs.html">PRScs</a></li>
<li class="toctree-l1"><a class="reference internal" href="PRScsx.html">PRScsx</a></li>
<li class="toctree-l1"><a class="reference internal" href="tlpSum.html">tlpSum</a></li>
<li class="toctree-l1"><a class="reference internal" href="PRSbils.html">PRSbils</a></li>
<li class="toctree-l1"><a class="reference internal" href="CTPR.html">CTPR</a></li>
<li class="toctree-l1"><a class="reference internal" href="NPS.html">NPS</a></li>
<li class="toctree-l1"><a class="reference internal" href="SDPR.html">SDPR</a></li>
<li class="toctree-l1"><a class="reference internal" href="JAMPred.html">JAMPred</a></li>
<li class="toctree-l1"><a class="reference internal" href="EBPRS.html">EB-PRS</a></li>
<li class="toctree-l1"><a class="reference internal" href="PANPRS.html">PANPRS</a></li>
<li class="toctree-l1"><a class="reference internal" href="BOLT-LMM.html">BOLT-LMM</a></li>
<li class="toctree-l1"><a class="reference internal" href="RapidoPGS-single.html">RapidoPGS-single</a></li>
<li class="toctree-l1"><a class="reference internal" href="LDpred-gibbs.html">LDpred-gibbs</a></li>
<li class="toctree-l1"><a class="reference internal" href="LDpred-fast.html">LDpred-fast</a></li>
<li class="toctree-l1"><a class="reference internal" href="LDpred-p%2Bt.html">LDpred-p+t</a></li>
<li class="toctree-l1"><a class="reference internal" href="LDpred-inf.html">LDpred-inf</a></li>
<li class="toctree-l1"><a class="reference internal" href="AnnoPredCode.html">AnnoPred</a></li>
<li class="toctree-l1"><a class="reference internal" href="PleioPredCode.html">PleioPred</a></li>
<li class="toctree-l1"><a class="reference internal" href="smtpredwMtOLS.html">smtpredwMtOLS</a></li>
<li class="toctree-l1"><a class="reference internal" href="smtpredwMtSBLUP.html">smtpredwMtSBLUP</a></li>
<li class="toctree-l1"><a class="reference internal" href="C%2BT.html">C+T</a></li>
<li class="toctree-l1"><a class="reference internal" href="viprs-simple.html">VIPRS-simple</a></li>
<li class="toctree-l1"><a class="reference internal" href="viprs-grid.html">VIPRS-grid</a></li>
<li class="toctree-l1"><a class="reference internal" href="HAIL.html">HAIL</a></li>
<li class="toctree-l1"><a class="reference internal" href="GEMMA-LM.html">GEMMA-LM</a></li>
<li class="toctree-l1"><a class="reference internal" href="GEMMA-LMM.html">GEMMA-LMM</a></li>
<li class="toctree-l1"><a class="reference internal" href="GEMMA-BSLMM.html">GEMMA-BSLMM</a></li>
<li class="toctree-l1"><a class="reference internal" href="MTG2.html">MTG2</a></li>
<li class="toctree-l1"><a class="reference internal" href="SCT.html">SCT</a></li>
<li class="toctree-l1"><a class="reference internal" href="XPBLUP.html">XPBLUP</a></li>
<li class="toctree-l1"><a class="reference internal" href="CTSLEB.html">CTSLEB</a></li>
<li class="toctree-l1"><a class="reference internal" href="PolyPred.html">PolyPred</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Polygenic Risk Score Tools</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Introduction - PRSTools</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/Introduction.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="introduction-prstools">
<h1>Introduction - PRSTools<a class="headerlink" href="#introduction-prstools" title="Link to this heading"></a></h1>
<section id="background">
<h2>Background<a class="headerlink" href="#background" title="Link to this heading"></a></h2>
<p>While working on a manuscript focused on benchmarking machine learning
algorithms for genotype-phenotype prediction, I received feedback
recommending the inclusion of a broader analysis of PRS methodologies.
This led me to investigate various PRS calculation tools used in genetic
research, each of which applies unique approaches and assumptions that
influence final PRS. These findings were supported by studies like <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/34243982/">this
one on PubMed</a>, which
reviews the diversity in PRS tools.</p>
<p>The calculation of PRS depends on several factors:</p>
<ul class="simple">
<li><p><strong>Data Type Compatibility</strong>: Some tools rely on GWAS summary
statistics, others use genotype data, and some utilize both types.</p></li>
<li><p><strong>Modeling Differences</strong>: Tools apply different mathematical models,
impacting PRS interpretation.</p></li>
<li><p><strong>Reference Panels and Genome Builds</strong>: PRS tools may require
specific reference panels, which affect compatibility and
generalizability.</p></li>
</ul>
<section id="key-challenges">
<h3>Key Challenges<a class="headerlink" href="#key-challenges" title="Link to this heading"></a></h3>
<p>Implementing PRS tools for practical analysis posed several challenges:</p>
<ul class="simple">
<li><p><strong>Data Format Incompatibility</strong>: Different tools accept varying input
formats, making data integration across tools challenging.</p></li>
<li><p><strong>Limited Cross-Validation Support</strong>: Many tools lack built-in
cross-validation functionality, essential for robust model
validation.</p></li>
<li><p><strong>HPC Scalability Constraints</strong>: Some tools are not optimized for
high-performance computing (HPC), limiting scalability for analyses
across multiple phenotypes.</p></li>
</ul>
</section>
</section>
<section id="repository-overview">
<h2>Repository Overview<a class="headerlink" href="#repository-overview" title="Link to this heading"></a></h2>
<p>To address these challenges, we created a new repository that provides a
unified implementation of PRS calculation tools with enhancements for
usability in real-world research settings. Key features include:</p>
<ol class="arabic simple">
<li><p><strong>Benchmarking</strong>: We benchmarked 46 PRS tools/methods on both binary
and continuous phenotypes.</p></li>
<li><p><strong>Comparative Analysis</strong>: Performance, computation time, memory
consumption, and beta distribution of each tool were compared.</p></li>
<li><p><strong>Data Transformation</strong>: Documentation on the necessary input data
transformations ensures compatibility across PRS tools.</p></li>
<li><p><strong>Parallel Execution</strong>: Tools were implemented for parallel execution
on HPC systems, enabling simultaneous analyses across multiple
phenotypes.</p></li>
<li><p><strong>Detailed Documentation</strong>: Comprehensive end-to-end documentation
for each PRS tool is available on GitHub.</p></li>
<li><p><strong>Diverse Dataset Testing</strong>: PRS tools were benchmarked on diverse
datasets, revealing specific tools’ limitations for certain data
types.</p></li>
<li><p><strong>Unified PRS Calculation Pipeline</strong>: A standardized evaluation
process was applied across all tools.</p></li>
<li><p><strong>Cross-Validation Implementation</strong>: Five-fold cross-validation was
incorporated to assess tool performance, an often-missing feature.</p></li>
<li><p><strong>Hyperparameter Tuning</strong>: PRS-specific hyperparameters (e.g.,
p-value thresholds, PCA counts) and tool-specific hyperparameters
were included to optimize PRS accuracy.</p></li>
</ol>
<figure class="align-default" id="id1">
<img alt="Introduction" src="_images/Introduction.png" />
<figcaption>
<p><span class="caption-text">Introduction</span><a class="headerlink" href="#id1" title="Link to this image"></a></p>
</figcaption>
</figure>
</section>
<section id="conclusion">
<h2>Conclusion<a class="headerlink" href="#conclusion" title="Link to this heading"></a></h2>
<p>In summary, this repository addresses some limitations in current PRS
tools, providing researchers with an adaptable and efficient solution
for polygenic risk score calculations. By improving compatibility,
scalability, and documentation, the repository supports large-scale
genetic studies and promotes broader use of PRS methodologies in
research.</p>
<figure class="align-default" id="id2">
<img alt="Discussion.png" src="_images/Discussion.png" />
<figcaption>
<p><span class="caption-text">Discussion.png</span><a class="headerlink" href="#id2" title="Link to this image"></a></p>
</figcaption>
</figure>
</section>
<section id="prs-tools-included-in-the-analysis">
<h2>PRS Tools Included in the Analysis.<a class="headerlink" href="#prs-tools-included-in-the-analysis" title="Link to this heading"></a></h2>
<p>Following is the list of PRS tools included in the analysis.</p>
<p>Here’s the table with bare links:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>T
o
o
l</p></th>
<th class="head"><p>P
y
t
h
o
n</p></th>
<th class="head"><p>C
o
n
d
a
E
n
v
i
r
o
n
m
e
n
t</p></th>
<th class="head"><p>Tool
Link</p></th>
<th class="head"><p>P
heno
type
(Bin
ary/
Cont
inuo
us/B
oth)</p></th>
<th class="head"><p>Re
qui
res
Ge
not
ype
d
ata</p></th>
<th class="head"><p>R
e
q
u
i
r
e
s
G
W
A
S
d
a
t
a</p></th>
<th class="head"><p>Re
qu
ir
es
C
ov
ar
ia
te
da
ta</p></th>
<th class="head"><p>Re
qui
res
Ref
ere
nce
d
ata</p></th>
<th class="head"><p>D
epe
nde
nce</p></th>
<th class="head"><p>L
a
n
g
u
a
g
e</p></th>
<th class="head"><p>O
riginal
article
DOI</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>P
l
i
n
k</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>ht
tps://w
ww.cog-
genomic
s.org/p
link/1.
9/score</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>No</p></td>
<td><p>No</p></td>
<td><p>C</p></td>
<td><p>ht
tps://d
oi.org/
10.1086
/519795</p></td>
</tr>
<tr class="row-odd"><td><p>P
R
S
i
c
e
-
2</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>htt
ps://gi
thub.co
m/chois
hingwan
/PRSice</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>Op
ti
on
al</p></td>
<td><p>Opt
ion
al,
but
re
com
men
ded</p></td>
<td><p>No</p></td>
<td><ul class="simple">
<li></li>
</ul>
<p>a
n
d
R</p>
</td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
93/giga
science
/giz082</p></td>
</tr>
<tr class="row-even"><td><p>G
C
T
A</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>htt
ps://ya
nglab.w
estlake
.edu.cn
/softwa
re/gcta
/#SBLUP</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><ul class="simple">
<li></li>
</ul>
</td>
<td><p>h
ttps://
doi.org
/10.100
7/978-1
-62703-
447-0_9</p></td>
</tr>
<tr class="row-odd"><td><p>D
B
S
L
M
M</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>ht
tps://g
ithub.c
om/bios
tat0903
/DBSLMM</p></td>
<td><p>Both</p></td>
<td><p>No
(
opt
ion
al)</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Opt
ion
al,
but
re
com
men
ded</p></td>
<td><p>Pli
nk,
LD
pre
d-2</p></td>
<td><ul class="simple">
<li></li>
</ul>
<p>a
n
d
R</p>
</td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
16/j.aj
hg.2020
.03.013</p></td>
</tr>
<tr class="row-even"><td><p>l
a
s
s
o
s
u
m</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/rdrr.i
o/githu
b/tshma
k/lasso
sum/man
/lassos
um.html</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>R</p></td>
<td><p>https:
//doi.o
rg/10.1
002/gep
i.22050</p></td>
</tr>
<tr class="row-odd"><td><p>l
d
p
r
e
d
2
_
i
n
f</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>https:
//prive
fl.gith
ub.io/b
igsnpr/
article
s/LDpre
d2.html</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>R</p></td>
<td><p>https
://doi.
org/10.
1093/bi
oinform
atics/b
taa1029</p></td>
</tr>
<tr class="row-even"><td><p>l
d
p
r
e
d
2
_
g
r
i
d</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>https:
//prive
fl.gith
ub.io/b
igsnpr/
article
s/LDpre
d2.html</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>R</p></td>
<td><p>https
://doi.
org/10.
1093/bi
oinform
atics/b
taa1029</p></td>
</tr>
<tr class="row-odd"><td><p>l
d
p
r
e
d
2
_
a
u
t
o</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>https:
//prive
fl.gith
ub.io/b
igsnpr/
article
s/LDpre
d2.html</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>R</p></td>
<td><p>https
://doi.
org/10.
1093/bi
oinform
atics/b
taa1029</p></td>
</tr>
<tr class="row-even"><td><p>l
d
p
r
e
d
2
_
l
a
s
s
o
s
u
m
2</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>https:
//prive
fl.gith
ub.io/b
igsnpr/
article
s/LDpre
d2.html</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>R</p></td>
<td><p>https
://doi.
org/10.
1093/bi
oinform
atics/b
taa1029</p></td>
</tr>
<tr class="row-odd"><td><p>l
d
p
r
e
d
-
f
u
n
c
t</p></td>
<td><p>2</p></td>
<td><p>l
d
s
c
c</p></td>
<td><p>http
s://git
hub.com
/carlam
l/LDpre
d-funct</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pli
nk,
LD
pre
d-2</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s414
67-021-
25171-9</p></td>
</tr>
<tr class="row-even"><td><p>S
B
a
y
e
s
R</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>https
://cnsg
enomics
.com/so
ftware/
gctb/#D
ownload</p></td>
<td><p>Both</p></td>
<td><p>Yes
(LD
m
atr
ix)</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes
(LD
M
atr
ix)</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><ul class="simple">
<li></li>
</ul>
</td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s415
88-024-
01704-y</p></td>
</tr>
<tr class="row-odd"><td><p>S
B
a
y
e
s
R
C</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>https
://cnsg
enomics
.com/so
ftware/
gctb/#D
ownload</p></td>
<td><p>Both</p></td>
<td><p>Yes
(LD
m
atr
ix)</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes
(LD
M
atr
ix)</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><ul class="simple">
<li></li>
</ul>
</td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s414
67-019-
12653-0</p></td>
</tr>
<tr class="row-even"><td><p>L
D
A
K
-
g
e
n
o
t
y
p
e</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>http
s://dou
gspeed.
com/qui
ck-prs/</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>N
o</p></td>
<td><p>No</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><ul class="simple">
<li></li>
</ul>
</td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s414
67-021-
24485-y</p></td>
</tr>
<tr class="row-odd"><td><p>L
D
A
K
-
g
w
a
s</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>http
s://dou
gspeed.
com/qui
ck-prs/</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes
(Co
rre
lat
ion
M
atr
ix)</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><ul class="simple">
<li></li>
</ul>
</td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s414
67-021-
24485-y</p></td>
</tr>
<tr class="row-even"><td><p>P
R
S
c
s</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>https:
//githu
b.com/g
etian10
7/PRScs</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes
(LD
M
atr
ix)</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s414
67-019-
09718-5</p></td>
</tr>
<tr class="row-odd"><td><p>P
R
S
c
s
x</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/github
.com/ge
tian107
/PRScsx</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes
(LD
M
atr
ix)</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s415
88-022-
01054-7</p></td>
</tr>
<tr class="row-even"><td><p>t
l
p
S
u
m</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>h
ttps://
github.
com/jpa
ttee/pe
nRegSum</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>R</p></td>
<td><p>ht
tps://d
oi.org/
10.1371
/journa
l.pcbi.
1008271</p></td>
</tr>
<tr class="row-odd"><td><p>P
R
S
b
i
l
s</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>https
://gith
ub.com/
styvon/
PRSbils</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td></td>
</tr>
<tr class="row-even"><td><p>C
T
P
R</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>https:
//githu
b.com/w
onilchu
ng/CTPR</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><ul class="simple">
<li></li>
</ul>
</td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s414
67-019-
08535-0</p></td>
</tr>
<tr class="row-odd"><td><p>N
P
S</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>h
ttps://
github.
com/sgc
hun/nps</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>R</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
16/j.aj
hg.2020
.05.004</p></td>
</tr>
<tr class="row-even"><td><p>S
D
P
R</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>htt
ps://gi
thub.co
m/eldro
nzhou/S
DPR.git</p></td>
<td><p>Both</p></td>
<td><p>Yes
(LD
m
atr
ix)</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><ul class="simple">
<li></li>
</ul>
</td>
<td><p>ht
tps://d
oi.org/
10.1371
/journa
l.pgen.
1009697</p></td>
</tr>
<tr class="row-odd"><td><p>J
A
M
P
r
e
d</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>htt
ps://rd
rr.io/g
ithub/p
jnewcom
be/R2BG
LiMS/ma
n/JAMPr
ed.html</p></td>
<td><p>Both</p></td>
<td><p>No</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>R</p></td>
<td><p>https:
//doi.o
rg/10.1
002/gep
i.22245</p></td>
</tr>
<tr class="row-even"><td><p>E
B
-
P
R
S</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>http
s://git
hub.com
/shuang
song011
0/EBPRS</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>N
o</p></td>
<td><p>No</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>R</p></td>
<td><p>ht
tps://d
oi.org/
10.1371
/journa
l.pcbi.
1007565</p></td>
</tr>
<tr class="row-odd"><td><p>P
A
N
P
R
S</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p>https:
//githu
b.com/c
ran/PAN
PRSnext</p></td>
<td><p>Both</p></td>
<td><p>Yes
(LD
m
atr
ix)</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes
(LD
M
atr
ix)</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>R</p></td>
<td><p>htt
ps://do
i.org/1
0.1080/
0162145
9.2020.
1764849</p></td>
</tr>
<tr class="row-even"><td><p>B
O
L
T
-
L
M
M</p></td>
<td><p>3</p></td>
<td><p>g
e
n
e
t
i
c
s</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/alkesg
roup.br
oadinst
itute.o
rg/BOLT
-LMM/BO
<a href="#id6"><span class="problematic" id="id7">LT-LMM_</span></a>
manual.
html#x1
-470008</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>N
o</p></td>
<td><p>Y
es</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><ul class="simple">
<li></li>
</ul>
</td>
<td><p>htt
ps://do
i.org/1
0.1038/
ng.3190</p></td>
</tr>
<tr class="row-odd"><td><p>R
a
p
i
d
o
P
G
S
-
s
i
n
g
l
e</p></td>
<td><p>3</p></td>
<td><p>A
d
v
a
n
c
e
R</p></td>
<td><p>ht
tps://g
ithub.c
om/GRea
lesM/Ra
pidoPGS</p></td>
<td><p>Both</p></td>
<td><p>No</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>L
D
p
r
e
d
-
g
i
b
b
s</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td><p>https:
//githu
b.com/b
vilhjal
/ldpred</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
16/j.aj
hg.2015
.09.001</p></td>
</tr>
<tr class="row-odd"><td><p>L
D
p
r
e
d
-
p
+
t</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td><p>https:
//githu
b.com/b
vilhjal
/ldpred</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
16/j.aj
hg.2015
.09.001</p></td>
</tr>
<tr class="row-even"><td><p>L
D
p
r
e
d
-
i
n
f</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td><p>https:
//githu
b.com/b
vilhjal
/ldpred</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
16/j.aj
hg.2015
.09.001</p></td>
</tr>
<tr class="row-odd"><td><p>L
D
p
r
e
d
-
f
a
s
t</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td><p>https:
//githu
b.com/b
vilhjal
/ldpred</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
16/j.aj
hg.2015
.09.001</p></td>
</tr>
<tr class="row-even"><td><p>A
n
n
o
-
P
r
e
d</p></td>
<td><p>2</p></td>
<td><p>L
D
S
C
C</p></td>
<td><p>h
ttps://
github.
com/yim
inghu/A
nnoPred</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p>ht
tps://d
oi.org/
10.1371
/journa
l.pcbi.
1005589</p></td>
</tr>
<tr class="row-odd"><td><p>s
m
t
p
r
e
d
-
w
M
t
O
L
S</p></td>
<td><p>2</p></td>
<td><p>L
D
S
C
C</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/github
.com/uq
rmaie1/
smtpred</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s414
67-017-
02769-6</p></td>
</tr>
<tr class="row-even"><td><p>s
m
t
p
r
e
d
-
w
M
t
S
B
L
U
P</p></td>
<td><p>2</p></td>
<td><p>L
D
S
C
C</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/github
.com/uq
rmaie1/
smtpred</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s414
67-017-
02769-6</p></td>
</tr>
<tr class="row-odd"><td><p>C
l
u
m
p
i
n
g
a
n
d
T
h
r
e
s
h
o
l
d
i
n
g
)</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td></td>
</tr>
<tr class="row-even"><td><p>v
i
p
r
s
-
s
i
m
p
l
e</p></td>
<td><p>3</p></td>
<td><p>V
i
p
r
s
_
e
n
v</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/github
.com/sh
z9/vipr
s-paper</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
16/j.aj
hg.2023
.03.009</p></td>
</tr>
<tr class="row-odd"><td><p>v
i
p
r
s
-
g
r
i
d</p></td>
<td><p>3</p></td>
<td><p>V
i
p
r
s
_
e
n
v</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/github
.com/sh
z9/vipr
s-paper</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
16/j.aj
hg.2023
.03.009</p></td>
</tr>
<tr class="row-even"><td><p>H
A
I
L</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td><p>https:
//nbvie
wer.org
/github
/ddbj/i
mputati
on-serv
er-wf/b
lob/mai
n/Noteb
ooks/ha
il-prs-
tutoria
l.ipynb</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s415
88-023-
01648-9</p></td>
</tr>
<tr class="row-odd"><td><p>G
E
M
M
A
-
L
M</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td><p>ht
tps://g
ithub.c
om/gene
tics-st
atistic
s/GEMMA</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>Y
es</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><ul class="simple">
<li></li>
</ul>
</td>
<td><p>htt
ps://do
i.org/1
0.1038/
ng.2310</p></td>
</tr>
<tr class="row-even"><td><p>G
E
M
M
A
-
L
L
M</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td><p>ht
tps://g
ithub.c
om/gene
tics-st
atistic
s/GEMMA</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>Y
es</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><ul class="simple">
<li></li>
</ul>
</td>
<td><p>htt
ps://do
i.org/1
0.1038/
ng.2310</p></td>
</tr>
<tr class="row-odd"><td><p>G
E
M
M
A
_
B
S
L
M
M</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td><p>ht
tps://g
ithub.c
om/gene
tics-st
atistic
s/GEMMA</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>Y
es</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><ul class="simple">
<li></li>
</ul>
</td>
<td><p>htt
ps://do
i.org/1
0.1038/
ng.2310</p></td>
</tr>
<tr class="row-even"><td><p>M
T
G
2</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td><p>https
://site
s.googl
e.com/v
iew/s-h
ong-lee
-homepa
ge/mtg2</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>N
o</p></td>
<td><p>Y
es</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><ul class="simple">
<li></li>
</ul>
</td>
<td><p>htt
ps://do
i.org/1
0.1093/
bioinfo
rmatics
/btw012</p></td>
</tr>
<tr class="row-odd"><td><p>S
C
T</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td><p>ht
tps://p
rivefl.
github.
io/bigs
npr/art
icles/S
CT.html</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>R</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
16/j.aj
hg.2019
.11.001</p></td>
</tr>
<tr class="row-even"><td><p>X
P
-
B
L
U
P</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td><p>https:
//githu
b.com/t
anglab/
XP-BLUP</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>N
o</p></td>
<td><p>No</p></td>
<td><p>No</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>B
a
s
h</p></td>
<td><p>http
s://doi
.org/10
.5808/g
i.21053</p></td>
</tr>
<tr class="row-odd"><td><p>C
T
S
L
E
B</p></td>
<td><p>3</p></td>
<td><p>G
e
n
e
t
i
c
s</p></td>
<td><p>ht
tps://g
ithub.c
om/andr
ewhaoyu
/CTSLEB</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>R</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s415
88-023-
01501-z</p></td>
</tr>
<tr class="row-even"><td><p>P
o
l
y
P
r
e
d</p></td>
<td><p>3</p></td>
<td><p>P
o
l
y
f
u
n</p></td>
<td><p>htt
ps://gi
thub.co
m/omerw
e/polyf
un/wiki
/6.-Tra
ns-ethn
ic-poly
genic-r
isk-pre
diction
-with-P
olyPred</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p><a class="reference external" href="https:/">https:/</a>
/doi.or
g/10.10
38/s415
88-022-
01036-9</p></td>
</tr>
<tr class="row-odd"><td><p>P
l
e
i
o
-
P
r
e
d</p></td>
<td><p>2</p></td>
<td><p>L
D
S
C
C</p></td>
<td><p>ht
tps://g
ithub.c
om/yimi
nghu/Pl
eioPred</p></td>
<td><p>Both</p></td>
<td><p>Yes</p></td>
<td><p>Y
e
s</p></td>
<td><p>No</p></td>
<td><p>Yes</p></td>
<td><p>Pl
ink
(
PRS
cal
cul
ati
on)</p></td>
<td><p>P
y
t
h
o
n</p></td>
<td><p>ht
tps://d
oi.org/
10.1371
/journa
l.pgen.
1006836</p></td>
</tr>
</tbody>
</table>
</section>
<section id="conda-environment">
<h2>Conda Environment<a class="headerlink" href="#conda-environment" title="Link to this heading"></a></h2>
<p>You may need to create the following Conda Environment to execute each
code</p>
<ul>
<li><div class="line-block">
<div class="line"><strong>advanceR Environment</strong></div>
<div class="line"><a class="reference external" href="CondaEnvironmentsForPRSTools/advanceR/environment.yml">Download
environment.yml</a></div>
<div class="line"><a class="reference external" href="CondaEnvironmentsForPRSTools/advanceR/replication_instructions.txt">Download
replication_instructions.txt</a></div>
</div>
</li>
<li><div class="line-block">
<div class="line"><strong>genetics Environment</strong></div>
<div class="line"><a class="reference external" href="CondaEnvironmentsForPRSTools/genetics/environment.yml">Download
environment.yml</a></div>
<div class="line"><a class="reference external" href="CondaEnvironmentsForPRSTools/genetics/replication_instructions.txt">Download
replication_instructions.txt</a></div>
</div>
</li>
<li><div class="line-block">
<div class="line"><strong>ldscc Environment</strong></div>
<div class="line"><a class="reference external" href="CondaEnvironmentsForPRSTools/ldscc/environment.yml">Download
environment.yml</a></div>
<div class="line"><a class="reference external" href="CondaEnvironmentsForPRSTools/ldscc/replication_instructions.txt">Download
replication_instructions.txt</a></div>
</div>
</li>
<li><div class="line-block">
<div class="line"><strong>polyfun Environment</strong></div>
<div class="line"><a class="reference external" href="CondaEnvironmentsForPRSTools/polyfun/environment.yml">Download
environment.yml</a></div>
<div class="line"><a class="reference external" href="CondaEnvironmentsForPRSTools/polyfun/replication_instructions.txt">Download
replication_instructions.txt</a></div>
</div>
</li>
<li><div class="line-block">
<div class="line"><strong>viprs_env Environment</strong></div>
<div class="line"><a class="reference external" href="CondaEnvironmentsForPRSTools/viprs_env/environment.yml">Download
environment.yml</a></div>
<div class="line"><a class="reference external" href="CondaEnvironmentsForPRSTools/viprs_env/replication_instructions.txt">Download
replication_instructions.txt</a></div>
</div>
</li>
</ul>
<p>The following tools were discarded from further consideration for the
following reasons:</p>
<ul class="simple">
<li><p><strong>Multiprs</strong> – This methodology calculates PRS for multiple p-value
thresholds and then combines them to form a single prediction. It is
a methodolgy rather a tool.</p></li>
<li><p><strong>BGLR-R</strong> – While this software worked with its provided test data,
it failed with our data, returning <code class="docutils literal notranslate"><span class="pre">NaN</span></code> for the explained
variances across all phenotypes. This was despite our data matching
the required format, with no missing genotype values for SNPs or
individuals.</p></li>
<li><p><strong>PolyRiskScore</strong> – This is a web-based tool that calculates PRS for
specific SNPs and uses GWAS files from the GWAS catalog. Due to its
limited flexibility and dependency on specific SNPs, it was not
suitable for our needs.</p></li>
<li><p><strong>FairPRS</strong> – Although promising, this tool writes output files to
the same directory, making it incompatible with running multiple
phenotypes or datasets simultaneously on HPC. The lack of parallel
processing capability led us to exclude it from further
consideration.</p></li>
<li><p><strong>RapidoPGS-multi</strong> – This tool did not support continuous
phenotypes, though it worked for binary ones. Due to this limitation,
it was removed from further consideration.</p></li>
</ul>
</section>
<section id="results">
<h2>Results<a class="headerlink" href="#results" title="Link to this heading"></a></h2>
<figure class="align-default" id="id3">
<img alt="TableResults.png" src="_images/TableResults.png" />
<figcaption>
<p><span class="caption-text">TableResults.png</span><a class="headerlink" href="#id3" title="Link to this image"></a></p>
</figcaption>
</figure>
</section>
<section id="related-projects">
<h2>Related Projects.<a class="headerlink" href="#related-projects" title="Link to this heading"></a></h2>
<p>Here are some of the related projects that would be very helpful to
researchers working on genetics, genotype-phenotype prediction, and risk
scores.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><ul class="simple">
<li></li>
</ul>
<p><em>Project
Title*</em></p>
</th>
<th class="head"><p><strong>Description</strong></p></th>
<th class="head"><p><strong>GitHub
Link</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><strong>G
WASPoker
forPRS</strong></p></td>
<td><p>A meta-analysis and parsing tool designed
for efficient polygenic risk score
calculation using GWAS summary statistic
files. This tool scans the GWAS Catalog,
downloads metadata, parses GWAS files, and
extracts necessary columns for PRS
calculation, including DOI and citation
details, helping to streamline the PRS
calculation process.</p></td>
<td><p><a class="reference external" href="https://github.com/MuhammadMuneeb007/GWASPokerforPRS">GitHub
Link</a></p></td>
</tr>
<tr class="row-odd"><td><p><strong>Ide
ntifying
Genes
As
sociated
with
Ph
enotypes
Using
Machine
and Deep
Le
arning</strong></p></td>
<td><p>A project focused on leveraging machine and
deep learning to identify genes associated
with various phenotypes. This approach aims
to uncover genetic associations that could
enhance our understanding of
phenotype-genotype relationships, using
advanced computational techniques.</p></td>
<td><p><a href="#id4"><span class="problematic" id="id5">`GitHub
Link &lt;
https://g
ithub.com
/Muhammad
Muneeb007
/Identify
ing-genes
-associat
ed-with-3
0-phenoty
pes-using
-machine-
deep-lear
ning-&gt;`__</span></a></p></td>
</tr>
<tr class="row-even"><td><p><strong>Benc
hmarking
80
OpenSNP
Ph
enotypes
Using
Deep
Learning
Al
gorithms
and PRS
Tools</strong></p></td>
<td><p>This project benchmarks 80 phenotypes from
OpenSNP using deep learning algorithms and
PRS tools. It aims to evaluate performance
across various methods, providing insights
into the effectiveness of different
algorithms for PRS computation and phenotype
prediction.</p></td>
<td><p><a class="reference external" href="https://github.com/MuhammadMuneeb007/Benchmarking-80-OpenSNP-phenotypes-using-deep-learning-algorithms-and-polygenic-risk-scores-tools">GitHub
Link</a></p></td>
</tr>
<tr class="row-odd"><td><p><strong>Heri
tability
Tools</strong></p></td>
<td><p>A collection of tools for calculating
heritability using diverse statistical
methods and datasets, including GWAS summary
statistics, genotype data, covariates, PCA,
and reference panels. This project serves as
a resource for researchers to estimate
heritability under different modeling
assumptions and statistical approaches.</p></td>
<td><p><a class="reference external" href="https://github.com/MuhammadMuneeb007/heritability">GitHub
Link</a></p></td>
</tr>
</tbody>
</table>
</section>
<section id="author-information">
<h2>Author Information.<a class="headerlink" href="#author-information" title="Link to this heading"></a></h2>
<ul class="simple">
<li><p><strong>Name</strong>: Muhammad Muneeb</p></li>
<li><p><strong>Affiliation</strong>: The University of Queensland</p></li>
<li><p><strong>Email</strong>: <a class="reference external" href="mailto:m&#46;muneeb&#37;&#52;&#48;uq&#46;edu&#46;au">m<span>&#46;</span>muneeb<span>&#64;</span>uq<span>&#46;</span>edu<span>&#46;</span>au</a></p></li>
<li><p><strong>Gmail</strong>: <a class="reference external" href="mailto:muneebsiddique007&#37;&#52;&#48;gmail&#46;com">muneebsiddique007<span>&#64;</span>gmail<span>&#46;</span>com</a></p></li>
<li><p><strong>GitHub</strong>: <a class="reference external" href="https://github.com/MuhammadMuneeb007/">GitHub
Profile</a></p></li>
<li><p><strong>Google Scholar</strong>: <a class="reference external" href="https://scholar.google.com/citations?hl=en&amp;user=X0xdltIAAAAJ&amp;view_op=list_works&amp;sortby=pubdate">Google
Scholar</a></p></li>
<li><p><strong>ResearchGate</strong>: <a class="reference external" href="https://www.researchgate.net/profile/Muhammad-Muneeb-5">ResearchGate
Profile</a></p></li>
<li><p><strong>Supervisor</strong>: <a class="reference external" href="https://scmb.uq.edu.au/profile/8654/david-ascher">David
Ascher</a></p></li>
<li><p><strong>Group Webpage</strong>: <a class="reference external" href="https://biosig.lab.uq.edu.au/">BioSig Lab</a></p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="index.html" class="btn btn-neutral float-left" title="Welcome to Polygenic Risk Score Tools’s documentation!" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="Step1-DownloadSampleDataandSetupEnvironment.html" class="btn btn-neutral float-right" title="Step1-DownloadSampleDataandSetupEnvironment" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Muhammad Muneeb.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>